// @azure/arm-policy/dist/esm/api/policyDefinitions/operations.js
import {
  errorResponseDeserializer,
  policyDefinitionSerializer,
  policyDefinitionDeserializer,
  _policyDefinitionListResultDeserializer
} from "../../models/models.js";
import { buildPagedAsyncIterator } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
function _listByManagementGroupSend(context, managementGroupId, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyDefinitions{?api%2Dversion,%24filter,%24top}",
    {
      managementGroupId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listByManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyDefinitionListResultDeserializer(result.body);
}
function listByManagementGroup(context, managementGroupId, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listByManagementGroupSend(context, managementGroupId, options),
    _listByManagementGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _deleteAtManagementGroupSend(context, managementGroupId, policyDefinitionName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      managementGroupId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
async function _deleteAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return;
}
async function deleteAtManagementGroup(context, managementGroupId, policyDefinitionName, options = { requestOptions: {} }) {
  const result = await _deleteAtManagementGroupSend(
    context,
    managementGroupId,
    policyDefinitionName,
    options
  );
  return _deleteAtManagementGroupDeserialize(result);
}
function _createOrUpdateAtManagementGroupSend(context, managementGroupId, policyDefinitionName, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      managementGroupId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyDefinitionSerializer(parameters)
  });
}
async function _createOrUpdateAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyDefinitionDeserializer(result.body);
}
async function createOrUpdateAtManagementGroup(context, managementGroupId, policyDefinitionName, parameters, options = { requestOptions: {} }) {
  const result = await _createOrUpdateAtManagementGroupSend(
    context,
    managementGroupId,
    policyDefinitionName,
    parameters,
    options
  );
  return _createOrUpdateAtManagementGroupDeserialize(result);
}
function _getAtManagementGroupSend(context, managementGroupId, policyDefinitionName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      managementGroupId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyDefinitionDeserializer(result.body);
}
async function getAtManagementGroup(context, managementGroupId, policyDefinitionName, options = { requestOptions: {} }) {
  const result = await _getAtManagementGroupSend(
    context,
    managementGroupId,
    policyDefinitionName,
    options
  );
  return _getAtManagementGroupDeserialize(result);
}
function _listBuiltInSend(context, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Authorization/policyDefinitions{?api%2Dversion,%24filter,%24top}",
    {
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listBuiltInDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyDefinitionListResultDeserializer(result.body);
}
function listBuiltIn(context, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listBuiltInSend(context, options),
    _listBuiltInDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _getBuiltInSend(context, policyDefinitionName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getBuiltInDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyDefinitionDeserializer(result.body);
}
async function getBuiltIn(context, policyDefinitionName, options = { requestOptions: {} }) {
  const result = await _getBuiltInSend(context, policyDefinitionName, options);
  return _getBuiltInDeserialize(result);
}
function _listSend(context, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions{?api%2Dversion,%24filter,%24top}",
    {
      subscriptionId: context.subscriptionId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyDefinitionListResultDeserializer(result.body);
}
function list(context, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listSend(context, options),
    _listDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _$deleteSend(context, policyDefinitionName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
async function _$deleteDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return;
}
async function $delete(context, policyDefinitionName, options = { requestOptions: {} }) {
  const result = await _$deleteSend(context, policyDefinitionName, options);
  return _$deleteDeserialize(result);
}
function _createOrUpdateSend(context, policyDefinitionName, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyDefinitionSerializer(parameters)
  });
}
async function _createOrUpdateDeserialize(result) {
  const expectedStatuses = ["201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyDefinitionDeserializer(result.body);
}
async function createOrUpdate(context, policyDefinitionName, parameters, options = { requestOptions: {} }) {
  const result = await _createOrUpdateSend(context, policyDefinitionName, parameters, options);
  return _createOrUpdateDeserialize(result);
}
function _getSend(context, policyDefinitionName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyDefinitionDeserializer(result.body);
}
async function get(context, policyDefinitionName, options = { requestOptions: {} }) {
  const result = await _getSend(context, policyDefinitionName, options);
  return _getDeserialize(result);
}
export {
  $delete,
  _$deleteDeserialize,
  _$deleteSend,
  _createOrUpdateAtManagementGroupDeserialize,
  _createOrUpdateAtManagementGroupSend,
  _createOrUpdateDeserialize,
  _createOrUpdateSend,
  _deleteAtManagementGroupDeserialize,
  _deleteAtManagementGroupSend,
  _getAtManagementGroupDeserialize,
  _getAtManagementGroupSend,
  _getBuiltInDeserialize,
  _getBuiltInSend,
  _getDeserialize,
  _getSend,
  _listBuiltInDeserialize,
  _listBuiltInSend,
  _listByManagementGroupDeserialize,
  _listByManagementGroupSend,
  _listDeserialize,
  _listSend,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listBuiltIn,
  listByManagementGroup
};
