// @azure/arm-policy/dist/commonjs/models/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var models_exports = {};
__export(models_exports, {
  AssignmentType: () => import_models.AssignmentType,
  CreatedByType: () => import_models.CreatedByType,
  EnforcementMode: () => import_models.EnforcementMode,
  ErrorAdditionalInfo: () => import_models.ErrorAdditionalInfo,
  ErrorDetail: () => import_models.ErrorDetail,
  ErrorResponse: () => import_models.ErrorResponse,
  ExtensionResource: () => import_models.ExtensionResource,
  ExternalEndpointResult: () => import_models.ExternalEndpointResult,
  ExternalEvaluationEndpointInvocationResult: () => import_models.ExternalEvaluationEndpointInvocationResult,
  ExternalEvaluationEndpointSettings: () => import_models.ExternalEvaluationEndpointSettings,
  ExternalEvaluationEnforcementSettings: () => import_models.ExternalEvaluationEnforcementSettings,
  Identity: () => import_models.Identity,
  KnownAssignmentType: () => import_models.KnownAssignmentType,
  KnownCreatedByType: () => import_models.KnownCreatedByType,
  KnownEnforcementMode: () => import_models.KnownEnforcementMode,
  KnownExternalEndpointResult: () => import_models.KnownExternalEndpointResult,
  KnownOverrideKind: () => import_models.KnownOverrideKind,
  KnownParameterType: () => import_models.KnownParameterType,
  KnownPolicyTokenResult: () => import_models.KnownPolicyTokenResult,
  KnownPolicyType: () => import_models.KnownPolicyType,
  KnownSelectorKind: () => import_models.KnownSelectorKind,
  KnownVersions: () => import_models.KnownVersions,
  NonComplianceMessage: () => import_models.NonComplianceMessage,
  Override: () => import_models.Override,
  OverrideKind: () => import_models.OverrideKind,
  ParameterDefinitionsValue: () => import_models.ParameterDefinitionsValue,
  ParameterDefinitionsValueMetadata: () => import_models.ParameterDefinitionsValueMetadata,
  ParameterType: () => import_models.ParameterType,
  ParameterValuesValue: () => import_models.ParameterValuesValue,
  PolicyAssignment: () => import_models.PolicyAssignment,
  PolicyAssignmentProperties: () => import_models.PolicyAssignmentProperties,
  PolicyAssignmentUpdate: () => import_models.PolicyAssignmentUpdate,
  PolicyAssignmentUpdateProperties: () => import_models.PolicyAssignmentUpdateProperties,
  PolicyDefinition: () => import_models.PolicyDefinition,
  PolicyDefinitionGroup: () => import_models.PolicyDefinitionGroup,
  PolicyDefinitionProperties: () => import_models.PolicyDefinitionProperties,
  PolicyDefinitionReference: () => import_models.PolicyDefinitionReference,
  PolicyDefinitionVersion: () => import_models.PolicyDefinitionVersion,
  PolicyDefinitionVersionProperties: () => import_models.PolicyDefinitionVersionProperties,
  PolicyLogInfo: () => import_models.PolicyLogInfo,
  PolicySetDefinition: () => import_models.PolicySetDefinition,
  PolicySetDefinitionProperties: () => import_models.PolicySetDefinitionProperties,
  PolicySetDefinitionVersion: () => import_models.PolicySetDefinitionVersion,
  PolicySetDefinitionVersionProperties: () => import_models.PolicySetDefinitionVersionProperties,
  PolicyTokenOperation: () => import_models.PolicyTokenOperation,
  PolicyTokenRequest: () => import_models.PolicyTokenRequest,
  PolicyTokenResponse: () => import_models.PolicyTokenResponse,
  PolicyTokenResult: () => import_models.PolicyTokenResult,
  PolicyType: () => import_models.PolicyType,
  ProxyResource: () => import_models.ProxyResource,
  Resource: () => import_models.Resource,
  ResourceIdentityType: () => import_models.ResourceIdentityType,
  ResourceSelector: () => import_models.ResourceSelector,
  Selector: () => import_models.Selector,
  SelectorKind: () => import_models.SelectorKind,
  SystemData: () => import_models.SystemData,
  UserAssignedIdentitiesValue: () => import_models.UserAssignedIdentitiesValue
});
module.exports = __toCommonJS(models_exports);
var import_models = require("./models.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AssignmentType,
  CreatedByType,
  EnforcementMode,
  ErrorAdditionalInfo,
  ErrorDetail,
  ErrorResponse,
  ExtensionResource,
  ExternalEndpointResult,
  ExternalEvaluationEndpointInvocationResult,
  ExternalEvaluationEndpointSettings,
  ExternalEvaluationEnforcementSettings,
  Identity,
  KnownAssignmentType,
  KnownCreatedByType,
  KnownEnforcementMode,
  KnownExternalEndpointResult,
  KnownOverrideKind,
  KnownParameterType,
  KnownPolicyTokenResult,
  KnownPolicyType,
  KnownSelectorKind,
  KnownVersions,
  NonComplianceMessage,
  Override,
  OverrideKind,
  ParameterDefinitionsValue,
  ParameterDefinitionsValueMetadata,
  ParameterType,
  ParameterValuesValue,
  PolicyAssignment,
  PolicyAssignmentProperties,
  PolicyAssignmentUpdate,
  PolicyAssignmentUpdateProperties,
  PolicyDefinition,
  PolicyDefinitionGroup,
  PolicyDefinitionProperties,
  PolicyDefinitionReference,
  PolicyDefinitionVersion,
  PolicyDefinitionVersionProperties,
  PolicyLogInfo,
  PolicySetDefinition,
  PolicySetDefinitionProperties,
  PolicySetDefinitionVersion,
  PolicySetDefinitionVersionProperties,
  PolicyTokenOperation,
  PolicyTokenRequest,
  PolicyTokenResponse,
  PolicyTokenResult,
  PolicyType,
  ProxyResource,
  Resource,
  ResourceIdentityType,
  ResourceSelector,
  Selector,
  SelectorKind,
  SystemData,
  UserAssignedIdentitiesValue
});
