// @azure/arm-policy/dist/commonjs/models/models.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var models_exports = {};
__export(models_exports, {
  KnownAssignmentType: () => KnownAssignmentType,
  KnownCreatedByType: () => KnownCreatedByType,
  KnownEnforcementMode: () => KnownEnforcementMode,
  KnownExternalEndpointResult: () => KnownExternalEndpointResult,
  KnownOverrideKind: () => KnownOverrideKind,
  KnownParameterType: () => KnownParameterType,
  KnownPolicyTokenResult: () => KnownPolicyTokenResult,
  KnownPolicyType: () => KnownPolicyType,
  KnownSelectorKind: () => KnownSelectorKind,
  KnownVersions: () => KnownVersions,
  _policyAssignmentListResultDeserializer: () => _policyAssignmentListResultDeserializer,
  _policyAssignmentPropertiesDeserializer: () => _policyAssignmentPropertiesDeserializer,
  _policyAssignmentPropertiesSerializer: () => _policyAssignmentPropertiesSerializer,
  _policyAssignmentUpdatePropertiesSerializer: () => _policyAssignmentUpdatePropertiesSerializer,
  _policyDefinitionListResultDeserializer: () => _policyDefinitionListResultDeserializer,
  _policyDefinitionPropertiesDeserializer: () => _policyDefinitionPropertiesDeserializer,
  _policyDefinitionPropertiesSerializer: () => _policyDefinitionPropertiesSerializer,
  _policyDefinitionVersionListResultDeserializer: () => _policyDefinitionVersionListResultDeserializer,
  _policyDefinitionVersionPropertiesDeserializer: () => _policyDefinitionVersionPropertiesDeserializer,
  _policyDefinitionVersionPropertiesSerializer: () => _policyDefinitionVersionPropertiesSerializer,
  _policySetDefinitionListResultDeserializer: () => _policySetDefinitionListResultDeserializer,
  _policySetDefinitionPropertiesDeserializer: () => _policySetDefinitionPropertiesDeserializer,
  _policySetDefinitionPropertiesSerializer: () => _policySetDefinitionPropertiesSerializer,
  _policySetDefinitionVersionListResultDeserializer: () => _policySetDefinitionVersionListResultDeserializer,
  _policySetDefinitionVersionPropertiesDeserializer: () => _policySetDefinitionVersionPropertiesDeserializer,
  _policySetDefinitionVersionPropertiesSerializer: () => _policySetDefinitionVersionPropertiesSerializer,
  errorAdditionalInfoArrayDeserializer: () => errorAdditionalInfoArrayDeserializer,
  errorAdditionalInfoDeserializer: () => errorAdditionalInfoDeserializer,
  errorDetailArrayDeserializer: () => errorDetailArrayDeserializer,
  errorDetailDeserializer: () => errorDetailDeserializer,
  errorResponseDeserializer: () => errorResponseDeserializer,
  extensionResourceDeserializer: () => extensionResourceDeserializer,
  extensionResourceSerializer: () => extensionResourceSerializer,
  externalEvaluationEndpointInvocationResultArrayDeserializer: () => externalEvaluationEndpointInvocationResultArrayDeserializer,
  externalEvaluationEndpointInvocationResultDeserializer: () => externalEvaluationEndpointInvocationResultDeserializer,
  externalEvaluationEndpointSettingsDeserializer: () => externalEvaluationEndpointSettingsDeserializer,
  externalEvaluationEndpointSettingsSerializer: () => externalEvaluationEndpointSettingsSerializer,
  externalEvaluationEnforcementSettingsDeserializer: () => externalEvaluationEnforcementSettingsDeserializer,
  externalEvaluationEnforcementSettingsSerializer: () => externalEvaluationEnforcementSettingsSerializer,
  identityDeserializer: () => identityDeserializer,
  identitySerializer: () => identitySerializer,
  nonComplianceMessageArrayDeserializer: () => nonComplianceMessageArrayDeserializer,
  nonComplianceMessageArraySerializer: () => nonComplianceMessageArraySerializer,
  nonComplianceMessageDeserializer: () => nonComplianceMessageDeserializer,
  nonComplianceMessageSerializer: () => nonComplianceMessageSerializer,
  overrideArrayDeserializer: () => overrideArrayDeserializer,
  overrideArraySerializer: () => overrideArraySerializer,
  overrideDeserializer: () => overrideDeserializer,
  overrideSerializer: () => overrideSerializer,
  parameterDefinitionsValueDeserializer: () => parameterDefinitionsValueDeserializer,
  parameterDefinitionsValueMetadataDeserializer: () => parameterDefinitionsValueMetadataDeserializer,
  parameterDefinitionsValueMetadataSerializer: () => parameterDefinitionsValueMetadataSerializer,
  parameterDefinitionsValueRecordDeserializer: () => parameterDefinitionsValueRecordDeserializer,
  parameterDefinitionsValueRecordSerializer: () => parameterDefinitionsValueRecordSerializer,
  parameterDefinitionsValueSerializer: () => parameterDefinitionsValueSerializer,
  parameterValuesValueDeserializer: () => parameterValuesValueDeserializer,
  parameterValuesValueRecordDeserializer: () => parameterValuesValueRecordDeserializer,
  parameterValuesValueRecordSerializer: () => parameterValuesValueRecordSerializer,
  parameterValuesValueSerializer: () => parameterValuesValueSerializer,
  policyAssignmentArrayDeserializer: () => policyAssignmentArrayDeserializer,
  policyAssignmentArraySerializer: () => policyAssignmentArraySerializer,
  policyAssignmentDeserializer: () => policyAssignmentDeserializer,
  policyAssignmentPropertiesDeserializer: () => policyAssignmentPropertiesDeserializer,
  policyAssignmentPropertiesSerializer: () => policyAssignmentPropertiesSerializer,
  policyAssignmentSerializer: () => policyAssignmentSerializer,
  policyAssignmentUpdatePropertiesSerializer: () => policyAssignmentUpdatePropertiesSerializer,
  policyAssignmentUpdateSerializer: () => policyAssignmentUpdateSerializer,
  policyDefinitionArrayDeserializer: () => policyDefinitionArrayDeserializer,
  policyDefinitionArraySerializer: () => policyDefinitionArraySerializer,
  policyDefinitionDeserializer: () => policyDefinitionDeserializer,
  policyDefinitionGroupArrayDeserializer: () => policyDefinitionGroupArrayDeserializer,
  policyDefinitionGroupArraySerializer: () => policyDefinitionGroupArraySerializer,
  policyDefinitionGroupDeserializer: () => policyDefinitionGroupDeserializer,
  policyDefinitionGroupSerializer: () => policyDefinitionGroupSerializer,
  policyDefinitionPropertiesDeserializer: () => policyDefinitionPropertiesDeserializer,
  policyDefinitionPropertiesSerializer: () => policyDefinitionPropertiesSerializer,
  policyDefinitionReferenceArrayDeserializer: () => policyDefinitionReferenceArrayDeserializer,
  policyDefinitionReferenceArraySerializer: () => policyDefinitionReferenceArraySerializer,
  policyDefinitionReferenceDeserializer: () => policyDefinitionReferenceDeserializer,
  policyDefinitionReferenceSerializer: () => policyDefinitionReferenceSerializer,
  policyDefinitionSerializer: () => policyDefinitionSerializer,
  policyDefinitionVersionArrayDeserializer: () => policyDefinitionVersionArrayDeserializer,
  policyDefinitionVersionArraySerializer: () => policyDefinitionVersionArraySerializer,
  policyDefinitionVersionDeserializer: () => policyDefinitionVersionDeserializer,
  policyDefinitionVersionPropertiesDeserializer: () => policyDefinitionVersionPropertiesDeserializer,
  policyDefinitionVersionPropertiesSerializer: () => policyDefinitionVersionPropertiesSerializer,
  policyDefinitionVersionSerializer: () => policyDefinitionVersionSerializer,
  policyLogInfoDeserializer: () => policyLogInfoDeserializer,
  policySetDefinitionArrayDeserializer: () => policySetDefinitionArrayDeserializer,
  policySetDefinitionArraySerializer: () => policySetDefinitionArraySerializer,
  policySetDefinitionDeserializer: () => policySetDefinitionDeserializer,
  policySetDefinitionPropertiesDeserializer: () => policySetDefinitionPropertiesDeserializer,
  policySetDefinitionPropertiesSerializer: () => policySetDefinitionPropertiesSerializer,
  policySetDefinitionSerializer: () => policySetDefinitionSerializer,
  policySetDefinitionVersionArrayDeserializer: () => policySetDefinitionVersionArrayDeserializer,
  policySetDefinitionVersionArraySerializer: () => policySetDefinitionVersionArraySerializer,
  policySetDefinitionVersionDeserializer: () => policySetDefinitionVersionDeserializer,
  policySetDefinitionVersionPropertiesDeserializer: () => policySetDefinitionVersionPropertiesDeserializer,
  policySetDefinitionVersionPropertiesSerializer: () => policySetDefinitionVersionPropertiesSerializer,
  policySetDefinitionVersionSerializer: () => policySetDefinitionVersionSerializer,
  policyTokenOperationSerializer: () => policyTokenOperationSerializer,
  policyTokenRequestSerializer: () => policyTokenRequestSerializer,
  policyTokenResponseDeserializer: () => policyTokenResponseDeserializer,
  proxyResourceDeserializer: () => proxyResourceDeserializer,
  proxyResourceSerializer: () => proxyResourceSerializer,
  resourceDeserializer: () => resourceDeserializer,
  resourceSelectorArrayDeserializer: () => resourceSelectorArrayDeserializer,
  resourceSelectorArraySerializer: () => resourceSelectorArraySerializer,
  resourceSelectorDeserializer: () => resourceSelectorDeserializer,
  resourceSelectorSerializer: () => resourceSelectorSerializer,
  resourceSerializer: () => resourceSerializer,
  selectorArrayDeserializer: () => selectorArrayDeserializer,
  selectorArraySerializer: () => selectorArraySerializer,
  selectorDeserializer: () => selectorDeserializer,
  selectorSerializer: () => selectorSerializer,
  systemDataDeserializer: () => systemDataDeserializer,
  userAssignedIdentitiesValueDeserializer: () => userAssignedIdentitiesValueDeserializer,
  userAssignedIdentitiesValueRecordDeserializer: () => userAssignedIdentitiesValueRecordDeserializer,
  userAssignedIdentitiesValueRecordSerializer: () => userAssignedIdentitiesValueRecordSerializer,
  userAssignedIdentitiesValueSerializer: () => userAssignedIdentitiesValueSerializer
});
module.exports = __toCommonJS(models_exports);
var import_check_prop_undefined = require("../static-helpers/serialization/check-prop-undefined.js");
var import_serialize_record = require("../static-helpers/serialization/serialize-record.js");
function policyAssignmentSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, [
      "displayName",
      "policyDefinitionId",
      "definitionVersion",
      "notScopes",
      "parameters",
      "description",
      "metadata",
      "enforcementMode",
      "nonComplianceMessages",
      "resourceSelectors",
      "overrides",
      "assignmentType"
    ]) ? void 0 : _policyAssignmentPropertiesSerializer(item),
    location: item["location"],
    identity: !item["identity"] ? item["identity"] : identitySerializer(item["identity"])
  };
}
function policyAssignmentDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"]),
    ...!item["properties"] ? item["properties"] : _policyAssignmentPropertiesDeserializer(item["properties"]),
    location: item["location"],
    identity: !item["identity"] ? item["identity"] : identityDeserializer(item["identity"])
  };
}
function policyAssignmentPropertiesSerializer(item) {
  return {
    displayName: item["displayName"],
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    notScopes: !item["notScopes"] ? item["notScopes"] : item["notScopes"].map((p) => {
      return p;
    }),
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordSerializer(item["parameters"]),
    description: item["description"],
    metadata: item["metadata"],
    enforcementMode: item["enforcementMode"],
    nonComplianceMessages: !item["nonComplianceMessages"] ? item["nonComplianceMessages"] : nonComplianceMessageArraySerializer(item["nonComplianceMessages"]),
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArraySerializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
    assignmentType: item["assignmentType"]
  };
}
function policyAssignmentPropertiesDeserializer(item) {
  return {
    displayName: item["displayName"],
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    latestDefinitionVersion: item["latestDefinitionVersion"],
    effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
    scope: item["scope"],
    notScopes: !item["notScopes"] ? item["notScopes"] : item["notScopes"].map((p) => {
      return p;
    }),
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordDeserializer(item["parameters"]),
    description: item["description"],
    metadata: item["metadata"],
    enforcementMode: item["enforcementMode"],
    nonComplianceMessages: !item["nonComplianceMessages"] ? item["nonComplianceMessages"] : nonComplianceMessageArrayDeserializer(item["nonComplianceMessages"]),
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArrayDeserializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArrayDeserializer(item["overrides"]),
    assignmentType: item["assignmentType"],
    instanceId: item["instanceId"]
  };
}
function parameterValuesValueRecordSerializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : parameterValuesValueSerializer(item[key]);
  });
  return result;
}
function parameterValuesValueRecordDeserializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : parameterValuesValueDeserializer(item[key]);
  });
  return result;
}
function parameterValuesValueSerializer(item) {
  return { value: item["value"] };
}
function parameterValuesValueDeserializer(item) {
  return {
    value: item["value"]
  };
}
var KnownEnforcementMode = /* @__PURE__ */ ((KnownEnforcementMode2) => {
  KnownEnforcementMode2["Default"] = "Default";
  KnownEnforcementMode2["DoNotEnforce"] = "DoNotEnforce";
  KnownEnforcementMode2["Enroll"] = "Enroll";
  return KnownEnforcementMode2;
})(KnownEnforcementMode || {});
function nonComplianceMessageArraySerializer(result) {
  return result.map((item) => {
    return nonComplianceMessageSerializer(item);
  });
}
function nonComplianceMessageArrayDeserializer(result) {
  return result.map((item) => {
    return nonComplianceMessageDeserializer(item);
  });
}
function nonComplianceMessageSerializer(item) {
  return {
    message: item["message"],
    policyDefinitionReferenceId: item["policyDefinitionReferenceId"]
  };
}
function nonComplianceMessageDeserializer(item) {
  return {
    message: item["message"],
    policyDefinitionReferenceId: item["policyDefinitionReferenceId"]
  };
}
function resourceSelectorArraySerializer(result) {
  return result.map((item) => {
    return resourceSelectorSerializer(item);
  });
}
function resourceSelectorArrayDeserializer(result) {
  return result.map((item) => {
    return resourceSelectorDeserializer(item);
  });
}
function resourceSelectorSerializer(item) {
  return {
    name: item["name"],
    selectors: !item["selectors"] ? item["selectors"] : selectorArraySerializer(item["selectors"])
  };
}
function resourceSelectorDeserializer(item) {
  return {
    name: item["name"],
    selectors: !item["selectors"] ? item["selectors"] : selectorArrayDeserializer(item["selectors"])
  };
}
function selectorArraySerializer(result) {
  return result.map((item) => {
    return selectorSerializer(item);
  });
}
function selectorArrayDeserializer(result) {
  return result.map((item) => {
    return selectorDeserializer(item);
  });
}
function selectorSerializer(item) {
  return {
    kind: item["kind"],
    in: !item["in"] ? item["in"] : item["in"].map((p) => {
      return p;
    }),
    notIn: !item["notIn"] ? item["notIn"] : item["notIn"].map((p) => {
      return p;
    })
  };
}
function selectorDeserializer(item) {
  return {
    kind: item["kind"],
    in: !item["in"] ? item["in"] : item["in"].map((p) => {
      return p;
    }),
    notIn: !item["notIn"] ? item["notIn"] : item["notIn"].map((p) => {
      return p;
    })
  };
}
var KnownSelectorKind = /* @__PURE__ */ ((KnownSelectorKind2) => {
  KnownSelectorKind2["ResourceLocation"] = "resourceLocation";
  KnownSelectorKind2["ResourceType"] = "resourceType";
  KnownSelectorKind2["ResourceWithoutLocation"] = "resourceWithoutLocation";
  KnownSelectorKind2["PolicyDefinitionReferenceId"] = "policyDefinitionReferenceId";
  return KnownSelectorKind2;
})(KnownSelectorKind || {});
function overrideArraySerializer(result) {
  return result.map((item) => {
    return overrideSerializer(item);
  });
}
function overrideArrayDeserializer(result) {
  return result.map((item) => {
    return overrideDeserializer(item);
  });
}
function overrideSerializer(item) {
  return {
    kind: item["kind"],
    value: item["value"],
    selectors: !item["selectors"] ? item["selectors"] : selectorArraySerializer(item["selectors"])
  };
}
function overrideDeserializer(item) {
  return {
    kind: item["kind"],
    value: item["value"],
    selectors: !item["selectors"] ? item["selectors"] : selectorArrayDeserializer(item["selectors"])
  };
}
var KnownOverrideKind = /* @__PURE__ */ ((KnownOverrideKind2) => {
  KnownOverrideKind2["PolicyEffect"] = "policyEffect";
  KnownOverrideKind2["DefinitionVersion"] = "definitionVersion";
  return KnownOverrideKind2;
})(KnownOverrideKind || {});
var KnownAssignmentType = /* @__PURE__ */ ((KnownAssignmentType2) => {
  KnownAssignmentType2["NotSpecified"] = "NotSpecified";
  KnownAssignmentType2["System"] = "System";
  KnownAssignmentType2["SystemHidden"] = "SystemHidden";
  KnownAssignmentType2["Custom"] = "Custom";
  return KnownAssignmentType2;
})(KnownAssignmentType || {});
function identitySerializer(item) {
  return {
    type: item["type"],
    userAssignedIdentities: !item["userAssignedIdentities"] ? item["userAssignedIdentities"] : userAssignedIdentitiesValueRecordSerializer(item["userAssignedIdentities"])
  };
}
function identityDeserializer(item) {
  return {
    principalId: item["principalId"],
    tenantId: item["tenantId"],
    type: item["type"],
    userAssignedIdentities: !item["userAssignedIdentities"] ? item["userAssignedIdentities"] : userAssignedIdentitiesValueRecordDeserializer(item["userAssignedIdentities"])
  };
}
function userAssignedIdentitiesValueRecordSerializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : userAssignedIdentitiesValueSerializer(item[key]);
  });
  return result;
}
function userAssignedIdentitiesValueRecordDeserializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : userAssignedIdentitiesValueDeserializer(item[key]);
  });
  return result;
}
function userAssignedIdentitiesValueSerializer(item) {
  return item;
}
function userAssignedIdentitiesValueDeserializer(item) {
  return {
    principalId: item["principalId"],
    clientId: item["clientId"]
  };
}
function extensionResourceSerializer(item) {
  return item;
}
function extensionResourceDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"])
  };
}
function resourceSerializer(item) {
  return item;
}
function resourceDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"])
  };
}
function systemDataDeserializer(item) {
  return {
    createdBy: item["createdBy"],
    createdByType: item["createdByType"],
    createdAt: !item["createdAt"] ? item["createdAt"] : new Date(item["createdAt"]),
    lastModifiedBy: item["lastModifiedBy"],
    lastModifiedByType: item["lastModifiedByType"],
    lastModifiedAt: !item["lastModifiedAt"] ? item["lastModifiedAt"] : new Date(item["lastModifiedAt"])
  };
}
var KnownCreatedByType = /* @__PURE__ */ ((KnownCreatedByType2) => {
  KnownCreatedByType2["User"] = "User";
  KnownCreatedByType2["Application"] = "Application";
  KnownCreatedByType2["ManagedIdentity"] = "ManagedIdentity";
  KnownCreatedByType2["Key"] = "Key";
  return KnownCreatedByType2;
})(KnownCreatedByType || {});
function errorResponseDeserializer(item) {
  return {
    error: !item["error"] ? item["error"] : errorDetailDeserializer(item["error"])
  };
}
function errorDetailDeserializer(item) {
  return {
    code: item["code"],
    message: item["message"],
    target: item["target"],
    details: !item["details"] ? item["details"] : errorDetailArrayDeserializer(item["details"]),
    additionalInfo: !item["additionalInfo"] ? item["additionalInfo"] : errorAdditionalInfoArrayDeserializer(item["additionalInfo"])
  };
}
function errorDetailArrayDeserializer(result) {
  return result.map((item) => {
    return errorDetailDeserializer(item);
  });
}
function errorAdditionalInfoArrayDeserializer(result) {
  return result.map((item) => {
    return errorAdditionalInfoDeserializer(item);
  });
}
function errorAdditionalInfoDeserializer(item) {
  return {
    type: item["type"],
    info: item["info"]
  };
}
function policyAssignmentUpdateSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, ["resourceSelectors", "overrides"]) ? void 0 : _policyAssignmentUpdatePropertiesSerializer(item),
    location: item["location"],
    identity: !item["identity"] ? item["identity"] : identitySerializer(item["identity"])
  };
}
function policyAssignmentUpdatePropertiesSerializer(item) {
  return {
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArraySerializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"])
  };
}
function _policyAssignmentListResultDeserializer(item) {
  return {
    value: policyAssignmentArrayDeserializer(item["value"]),
    nextLink: item["nextLink"]
  };
}
function policyAssignmentArraySerializer(result) {
  return result.map((item) => {
    return policyAssignmentSerializer(item);
  });
}
function policyAssignmentArrayDeserializer(result) {
  return result.map((item) => {
    return policyAssignmentDeserializer(item);
  });
}
function policyDefinitionSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, [
      "policyType",
      "mode",
      "displayName",
      "description",
      "policyRule",
      "metadata",
      "parameters",
      "version",
      "versions",
      "externalEvaluationEnforcementSettings"
    ]) ? void 0 : _policyDefinitionPropertiesSerializer(item)
  };
}
function policyDefinitionDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"]),
    ...!item["properties"] ? item["properties"] : _policyDefinitionPropertiesDeserializer(item["properties"])
  };
}
function policyDefinitionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    }),
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsSerializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function policyDefinitionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    }),
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsDeserializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
var KnownPolicyType = /* @__PURE__ */ ((KnownPolicyType2) => {
  KnownPolicyType2["NotSpecified"] = "NotSpecified";
  KnownPolicyType2["BuiltIn"] = "BuiltIn";
  KnownPolicyType2["Custom"] = "Custom";
  KnownPolicyType2["Static"] = "Static";
  return KnownPolicyType2;
})(KnownPolicyType || {});
function parameterDefinitionsValueRecordSerializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : parameterDefinitionsValueSerializer(item[key]);
  });
  return result;
}
function parameterDefinitionsValueRecordDeserializer(item) {
  const result = {};
  Object.keys(item).map((key) => {
    result[key] = !item[key] ? item[key] : parameterDefinitionsValueDeserializer(item[key]);
  });
  return result;
}
function parameterDefinitionsValueSerializer(item) {
  return {
    type: item["type"],
    allowedValues: !item["allowedValues"] ? item["allowedValues"] : item["allowedValues"].map((p) => {
      return p;
    }),
    defaultValue: item["defaultValue"],
    schema: item["schema"],
    metadata: !item["metadata"] ? item["metadata"] : parameterDefinitionsValueMetadataSerializer(item["metadata"])
  };
}
function parameterDefinitionsValueDeserializer(item) {
  return {
    type: item["type"],
    allowedValues: !item["allowedValues"] ? item["allowedValues"] : item["allowedValues"].map((p) => {
      return p;
    }),
    defaultValue: item["defaultValue"],
    schema: item["schema"],
    metadata: !item["metadata"] ? item["metadata"] : parameterDefinitionsValueMetadataDeserializer(item["metadata"])
  };
}
var KnownParameterType = /* @__PURE__ */ ((KnownParameterType2) => {
  KnownParameterType2["String"] = "String";
  KnownParameterType2["Array"] = "Array";
  KnownParameterType2["Object"] = "Object";
  KnownParameterType2["Boolean"] = "Boolean";
  KnownParameterType2["Integer"] = "Integer";
  KnownParameterType2["Float"] = "Float";
  KnownParameterType2["DateTime"] = "DateTime";
  return KnownParameterType2;
})(KnownParameterType || {});
function parameterDefinitionsValueMetadataSerializer(item) {
  return {
    ...(0, import_serialize_record.serializeRecord)(item.additionalProperties ?? {}),
    displayName: item["displayName"],
    description: item["description"],
    strongType: item["strongType"],
    assignPermissions: item["assignPermissions"]
  };
}
function parameterDefinitionsValueMetadataDeserializer(item) {
  return {
    additionalProperties: (0, import_serialize_record.serializeRecord)(item, [
      "displayName",
      "description",
      "strongType",
      "assignPermissions"
    ]),
    displayName: item["displayName"],
    description: item["description"],
    strongType: item["strongType"],
    assignPermissions: item["assignPermissions"]
  };
}
function externalEvaluationEnforcementSettingsSerializer(item) {
  return {
    missingTokenAction: item["missingTokenAction"],
    resultLifespan: item["resultLifespan"],
    endpointSettings: !item["endpointSettings"] ? item["endpointSettings"] : externalEvaluationEndpointSettingsSerializer(item["endpointSettings"]),
    roleDefinitionIds: !item["roleDefinitionIds"] ? item["roleDefinitionIds"] : item["roleDefinitionIds"].map((p) => {
      return p;
    })
  };
}
function externalEvaluationEnforcementSettingsDeserializer(item) {
  return {
    missingTokenAction: item["missingTokenAction"],
    resultLifespan: item["resultLifespan"],
    endpointSettings: !item["endpointSettings"] ? item["endpointSettings"] : externalEvaluationEndpointSettingsDeserializer(item["endpointSettings"]),
    roleDefinitionIds: !item["roleDefinitionIds"] ? item["roleDefinitionIds"] : item["roleDefinitionIds"].map((p) => {
      return p;
    })
  };
}
function externalEvaluationEndpointSettingsSerializer(item) {
  return { kind: item["kind"], details: item["details"] };
}
function externalEvaluationEndpointSettingsDeserializer(item) {
  return {
    kind: item["kind"],
    details: item["details"]
  };
}
function proxyResourceSerializer(item) {
  return item;
}
function proxyResourceDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"])
  };
}
function _policyDefinitionListResultDeserializer(item) {
  return {
    value: policyDefinitionArrayDeserializer(item["value"]),
    nextLink: item["nextLink"]
  };
}
function policyDefinitionArraySerializer(result) {
  return result.map((item) => {
    return policyDefinitionSerializer(item);
  });
}
function policyDefinitionArrayDeserializer(result) {
  return result.map((item) => {
    return policyDefinitionDeserializer(item);
  });
}
function policyDefinitionVersionSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, [
      "policyType",
      "mode",
      "displayName",
      "description",
      "policyRule",
      "metadata",
      "parameters",
      "version",
      "externalEvaluationEnforcementSettings"
    ]) ? void 0 : _policyDefinitionVersionPropertiesSerializer(item)
  };
}
function policyDefinitionVersionDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"]),
    ...!item["properties"] ? item["properties"] : _policyDefinitionVersionPropertiesDeserializer(item["properties"])
  };
}
function policyDefinitionVersionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    version: item["version"],
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsSerializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function policyDefinitionVersionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    version: item["version"],
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsDeserializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function _policyDefinitionVersionListResultDeserializer(item) {
  return {
    value: policyDefinitionVersionArrayDeserializer(item["value"]),
    nextLink: item["nextLink"]
  };
}
function policyDefinitionVersionArraySerializer(result) {
  return result.map((item) => {
    return policyDefinitionVersionSerializer(item);
  });
}
function policyDefinitionVersionArrayDeserializer(result) {
  return result.map((item) => {
    return policyDefinitionVersionDeserializer(item);
  });
}
function policySetDefinitionSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, [
      "policyType",
      "displayName",
      "description",
      "metadata",
      "parameters",
      "policyDefinitions",
      "policyDefinitionGroups",
      "version",
      "versions"
    ]) ? void 0 : _policySetDefinitionPropertiesSerializer(item)
  };
}
function policySetDefinitionDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"]),
    ...!item["properties"] ? item["properties"] : _policySetDefinitionPropertiesDeserializer(item["properties"])
  };
}
function policySetDefinitionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    policyDefinitions: policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    })
  };
}
function policySetDefinitionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    policyDefinitions: policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    })
  };
}
function policyDefinitionReferenceArraySerializer(result) {
  return result.map((item) => {
    return policyDefinitionReferenceSerializer(item);
  });
}
function policyDefinitionReferenceArrayDeserializer(result) {
  return result.map((item) => {
    return policyDefinitionReferenceDeserializer(item);
  });
}
function policyDefinitionReferenceSerializer(item) {
  return {
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordSerializer(item["parameters"]),
    policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
    groupNames: !item["groupNames"] ? item["groupNames"] : item["groupNames"].map((p) => {
      return p;
    })
  };
}
function policyDefinitionReferenceDeserializer(item) {
  return {
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    latestDefinitionVersion: item["latestDefinitionVersion"],
    effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordDeserializer(item["parameters"]),
    policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
    groupNames: !item["groupNames"] ? item["groupNames"] : item["groupNames"].map((p) => {
      return p;
    })
  };
}
function policyDefinitionGroupArraySerializer(result) {
  return result.map((item) => {
    return policyDefinitionGroupSerializer(item);
  });
}
function policyDefinitionGroupArrayDeserializer(result) {
  return result.map((item) => {
    return policyDefinitionGroupDeserializer(item);
  });
}
function policyDefinitionGroupSerializer(item) {
  return {
    name: item["name"],
    displayName: item["displayName"],
    category: item["category"],
    description: item["description"],
    additionalMetadataId: item["additionalMetadataId"]
  };
}
function policyDefinitionGroupDeserializer(item) {
  return {
    name: item["name"],
    displayName: item["displayName"],
    category: item["category"],
    description: item["description"],
    additionalMetadataId: item["additionalMetadataId"]
  };
}
function _policySetDefinitionListResultDeserializer(item) {
  return {
    value: policySetDefinitionArrayDeserializer(item["value"]),
    nextLink: item["nextLink"]
  };
}
function policySetDefinitionArraySerializer(result) {
  return result.map((item) => {
    return policySetDefinitionSerializer(item);
  });
}
function policySetDefinitionArrayDeserializer(result) {
  return result.map((item) => {
    return policySetDefinitionDeserializer(item);
  });
}
function policySetDefinitionVersionSerializer(item) {
  return {
    properties: (0, import_check_prop_undefined.areAllPropsUndefined)(item, [
      "policyType",
      "displayName",
      "description",
      "metadata",
      "parameters",
      "policyDefinitions",
      "policyDefinitionGroups",
      "version"
    ]) ? void 0 : _policySetDefinitionVersionPropertiesSerializer(item)
  };
}
function policySetDefinitionVersionDeserializer(item) {
  return {
    id: item["id"],
    name: item["name"],
    type: item["type"],
    systemData: !item["systemData"] ? item["systemData"] : systemDataDeserializer(item["systemData"]),
    ...!item["properties"] ? item["properties"] : _policySetDefinitionVersionPropertiesDeserializer(item["properties"])
  };
}
function policySetDefinitionVersionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    policyDefinitions: policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
    version: item["version"]
  };
}
function policySetDefinitionVersionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    policyDefinitions: policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
    version: item["version"]
  };
}
function _policySetDefinitionVersionListResultDeserializer(item) {
  return {
    value: policySetDefinitionVersionArrayDeserializer(item["value"]),
    nextLink: item["nextLink"]
  };
}
function policySetDefinitionVersionArraySerializer(result) {
  return result.map((item) => {
    return policySetDefinitionVersionSerializer(item);
  });
}
function policySetDefinitionVersionArrayDeserializer(result) {
  return result.map((item) => {
    return policySetDefinitionVersionDeserializer(item);
  });
}
function policyTokenRequestSerializer(item) {
  return {
    operation: policyTokenOperationSerializer(item["operation"]),
    changeReference: item["changeReference"]
  };
}
function policyTokenOperationSerializer(item) {
  return { uri: item["uri"], httpMethod: item["httpMethod"], content: item["content"] };
}
function policyTokenResponseDeserializer(item) {
  return {
    result: item["result"],
    message: item["message"],
    retryAfter: !item["retryAfter"] ? item["retryAfter"] : new Date(item["retryAfter"]),
    results: !item["results"] ? item["results"] : externalEvaluationEndpointInvocationResultArrayDeserializer(item["results"]),
    changeReference: item["changeReference"],
    token: item["token"],
    tokenId: item["tokenId"],
    expiration: !item["expiration"] ? item["expiration"] : new Date(item["expiration"])
  };
}
var KnownPolicyTokenResult = /* @__PURE__ */ ((KnownPolicyTokenResult2) => {
  KnownPolicyTokenResult2["Succeeded"] = "Succeeded";
  KnownPolicyTokenResult2["Failed"] = "Failed";
  return KnownPolicyTokenResult2;
})(KnownPolicyTokenResult || {});
function externalEvaluationEndpointInvocationResultArrayDeserializer(result) {
  return result.map((item) => {
    return externalEvaluationEndpointInvocationResultDeserializer(item);
  });
}
function externalEvaluationEndpointInvocationResultDeserializer(item) {
  return {
    policyInfo: !item["policyInfo"] ? item["policyInfo"] : policyLogInfoDeserializer(item["policyInfo"]),
    result: item["result"],
    message: item["message"],
    retryAfter: !item["retryAfter"] ? item["retryAfter"] : new Date(item["retryAfter"]),
    claims: item["claims"],
    expiration: !item["expiration"] ? item["expiration"] : new Date(item["expiration"])
  };
}
function policyLogInfoDeserializer(item) {
  return {
    policyDefinitionId: item["policyDefinitionId"],
    policySetDefinitionId: item["policySetDefinitionId"],
    policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
    policySetDefinitionName: item["policySetDefinitionName"],
    policySetDefinitionDisplayName: item["policySetDefinitionDisplayName"],
    policySetDefinitionVersion: item["policySetDefinitionVersion"],
    policySetDefinitionCategory: item["policySetDefinitionCategory"],
    policyDefinitionName: item["policyDefinitionName"],
    policyDefinitionDisplayName: item["policyDefinitionDisplayName"],
    policyDefinitionVersion: item["policyDefinitionVersion"],
    policyDefinitionEffect: item["policyDefinitionEffect"],
    policyDefinitionGroupNames: !item["policyDefinitionGroupNames"] ? item["policyDefinitionGroupNames"] : item["policyDefinitionGroupNames"].map((p) => {
      return p;
    }),
    policyAssignmentId: item["policyAssignmentId"],
    policyAssignmentName: item["policyAssignmentName"],
    policyAssignmentDisplayName: item["policyAssignmentDisplayName"],
    policyAssignmentVersion: item["policyAssignmentVersion"],
    policyAssignmentScope: item["policyAssignmentScope"],
    resourceLocation: item["resourceLocation"],
    ancestors: item["ancestors"],
    complianceReasonCode: item["complianceReasonCode"],
    policyExemptionIds: !item["policyExemptionIds"] ? item["policyExemptionIds"] : item["policyExemptionIds"].map((p) => {
      return p;
    })
  };
}
var KnownExternalEndpointResult = /* @__PURE__ */ ((KnownExternalEndpointResult2) => {
  KnownExternalEndpointResult2["Succeeded"] = "Succeeded";
  KnownExternalEndpointResult2["Failed"] = "Failed";
  return KnownExternalEndpointResult2;
})(KnownExternalEndpointResult || {});
var KnownVersions = /* @__PURE__ */ ((KnownVersions2) => {
  KnownVersions2["V20250301"] = "2025-03-01";
  return KnownVersions2;
})(KnownVersions || {});
function _policyAssignmentPropertiesSerializer(item) {
  return {
    displayName: item["displayName"],
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    notScopes: !item["notScopes"] ? item["notScopes"] : item["notScopes"].map((p) => {
      return p;
    }),
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordSerializer(item["parameters"]),
    description: item["description"],
    metadata: item["metadata"],
    enforcementMode: item["enforcementMode"],
    nonComplianceMessages: !item["nonComplianceMessages"] ? item["nonComplianceMessages"] : nonComplianceMessageArraySerializer(item["nonComplianceMessages"]),
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArraySerializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
    assignmentType: item["assignmentType"]
  };
}
function _policyAssignmentPropertiesDeserializer(item) {
  return {
    displayName: item["displayName"],
    policyDefinitionId: item["policyDefinitionId"],
    definitionVersion: item["definitionVersion"],
    latestDefinitionVersion: item["latestDefinitionVersion"],
    effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
    scope: item["scope"],
    notScopes: !item["notScopes"] ? item["notScopes"] : item["notScopes"].map((p) => {
      return p;
    }),
    parameters: !item["parameters"] ? item["parameters"] : parameterValuesValueRecordDeserializer(item["parameters"]),
    description: item["description"],
    metadata: item["metadata"],
    enforcementMode: item["enforcementMode"],
    nonComplianceMessages: !item["nonComplianceMessages"] ? item["nonComplianceMessages"] : nonComplianceMessageArrayDeserializer(item["nonComplianceMessages"]),
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArrayDeserializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArrayDeserializer(item["overrides"]),
    assignmentType: item["assignmentType"],
    instanceId: item["instanceId"]
  };
}
function _policyAssignmentUpdatePropertiesSerializer(item) {
  return {
    resourceSelectors: !item["resourceSelectors"] ? item["resourceSelectors"] : resourceSelectorArraySerializer(item["resourceSelectors"]),
    overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"])
  };
}
function _policyDefinitionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    }),
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsSerializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function _policyDefinitionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    }),
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsDeserializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function _policyDefinitionVersionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    version: item["version"],
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsSerializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function _policyDefinitionVersionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    mode: item["mode"],
    displayName: item["displayName"],
    description: item["description"],
    policyRule: item["policyRule"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    version: item["version"],
    externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"] ? item["externalEvaluationEnforcementSettings"] : externalEvaluationEnforcementSettingsDeserializer(
      item["externalEvaluationEnforcementSettings"]
    )
  };
}
function _policySetDefinitionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    policyDefinitions: !item["policyDefinitions"] ? item["policyDefinitions"] : policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    })
  };
}
function _policySetDefinitionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    policyDefinitions: !item["policyDefinitions"] ? item["policyDefinitions"] : policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
    version: item["version"],
    versions: !item["versions"] ? item["versions"] : item["versions"].map((p) => {
      return p;
    })
  };
}
function _policySetDefinitionVersionPropertiesSerializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordSerializer(item["parameters"]),
    policyDefinitions: !item["policyDefinitions"] ? item["policyDefinitions"] : policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
    version: item["version"]
  };
}
function _policySetDefinitionVersionPropertiesDeserializer(item) {
  return {
    policyType: item["policyType"],
    displayName: item["displayName"],
    description: item["description"],
    metadata: item["metadata"],
    parameters: !item["parameters"] ? item["parameters"] : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
    policyDefinitions: !item["policyDefinitions"] ? item["policyDefinitions"] : policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
    policyDefinitionGroups: !item["policyDefinitionGroups"] ? item["policyDefinitionGroups"] : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
    version: item["version"]
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  KnownAssignmentType,
  KnownCreatedByType,
  KnownEnforcementMode,
  KnownExternalEndpointResult,
  KnownOverrideKind,
  KnownParameterType,
  KnownPolicyTokenResult,
  KnownPolicyType,
  KnownSelectorKind,
  KnownVersions,
  _policyAssignmentListResultDeserializer,
  _policyAssignmentPropertiesDeserializer,
  _policyAssignmentPropertiesSerializer,
  _policyAssignmentUpdatePropertiesSerializer,
  _policyDefinitionListResultDeserializer,
  _policyDefinitionPropertiesDeserializer,
  _policyDefinitionPropertiesSerializer,
  _policyDefinitionVersionListResultDeserializer,
  _policyDefinitionVersionPropertiesDeserializer,
  _policyDefinitionVersionPropertiesSerializer,
  _policySetDefinitionListResultDeserializer,
  _policySetDefinitionPropertiesDeserializer,
  _policySetDefinitionPropertiesSerializer,
  _policySetDefinitionVersionListResultDeserializer,
  _policySetDefinitionVersionPropertiesDeserializer,
  _policySetDefinitionVersionPropertiesSerializer,
  errorAdditionalInfoArrayDeserializer,
  errorAdditionalInfoDeserializer,
  errorDetailArrayDeserializer,
  errorDetailDeserializer,
  errorResponseDeserializer,
  extensionResourceDeserializer,
  extensionResourceSerializer,
  externalEvaluationEndpointInvocationResultArrayDeserializer,
  externalEvaluationEndpointInvocationResultDeserializer,
  externalEvaluationEndpointSettingsDeserializer,
  externalEvaluationEndpointSettingsSerializer,
  externalEvaluationEnforcementSettingsDeserializer,
  externalEvaluationEnforcementSettingsSerializer,
  identityDeserializer,
  identitySerializer,
  nonComplianceMessageArrayDeserializer,
  nonComplianceMessageArraySerializer,
  nonComplianceMessageDeserializer,
  nonComplianceMessageSerializer,
  overrideArrayDeserializer,
  overrideArraySerializer,
  overrideDeserializer,
  overrideSerializer,
  parameterDefinitionsValueDeserializer,
  parameterDefinitionsValueMetadataDeserializer,
  parameterDefinitionsValueMetadataSerializer,
  parameterDefinitionsValueRecordDeserializer,
  parameterDefinitionsValueRecordSerializer,
  parameterDefinitionsValueSerializer,
  parameterValuesValueDeserializer,
  parameterValuesValueRecordDeserializer,
  parameterValuesValueRecordSerializer,
  parameterValuesValueSerializer,
  policyAssignmentArrayDeserializer,
  policyAssignmentArraySerializer,
  policyAssignmentDeserializer,
  policyAssignmentPropertiesDeserializer,
  policyAssignmentPropertiesSerializer,
  policyAssignmentSerializer,
  policyAssignmentUpdatePropertiesSerializer,
  policyAssignmentUpdateSerializer,
  policyDefinitionArrayDeserializer,
  policyDefinitionArraySerializer,
  policyDefinitionDeserializer,
  policyDefinitionGroupArrayDeserializer,
  policyDefinitionGroupArraySerializer,
  policyDefinitionGroupDeserializer,
  policyDefinitionGroupSerializer,
  policyDefinitionPropertiesDeserializer,
  policyDefinitionPropertiesSerializer,
  policyDefinitionReferenceArrayDeserializer,
  policyDefinitionReferenceArraySerializer,
  policyDefinitionReferenceDeserializer,
  policyDefinitionReferenceSerializer,
  policyDefinitionSerializer,
  policyDefinitionVersionArrayDeserializer,
  policyDefinitionVersionArraySerializer,
  policyDefinitionVersionDeserializer,
  policyDefinitionVersionPropertiesDeserializer,
  policyDefinitionVersionPropertiesSerializer,
  policyDefinitionVersionSerializer,
  policyLogInfoDeserializer,
  policySetDefinitionArrayDeserializer,
  policySetDefinitionArraySerializer,
  policySetDefinitionDeserializer,
  policySetDefinitionPropertiesDeserializer,
  policySetDefinitionPropertiesSerializer,
  policySetDefinitionSerializer,
  policySetDefinitionVersionArrayDeserializer,
  policySetDefinitionVersionArraySerializer,
  policySetDefinitionVersionDeserializer,
  policySetDefinitionVersionPropertiesDeserializer,
  policySetDefinitionVersionPropertiesSerializer,
  policySetDefinitionVersionSerializer,
  policyTokenOperationSerializer,
  policyTokenRequestSerializer,
  policyTokenResponseDeserializer,
  proxyResourceDeserializer,
  proxyResourceSerializer,
  resourceDeserializer,
  resourceSelectorArrayDeserializer,
  resourceSelectorArraySerializer,
  resourceSelectorDeserializer,
  resourceSelectorSerializer,
  resourceSerializer,
  selectorArrayDeserializer,
  selectorArraySerializer,
  selectorDeserializer,
  selectorSerializer,
  systemDataDeserializer,
  userAssignedIdentitiesValueDeserializer,
  userAssignedIdentitiesValueRecordDeserializer,
  userAssignedIdentitiesValueRecordSerializer,
  userAssignedIdentitiesValueSerializer
});
