// @azure/notification-hubs/dist/commonjs/models/notificationBodyBuilder.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notificationBodyBuilder_exports = {};
__export(notificationBodyBuilder_exports, {
  createAdmNotificationBody: () => createAdmNotificationBody,
  createAppleNotificationBody: () => createAppleNotificationBody,
  createBaiduNotificationBody: () => createBaiduNotificationBody,
  createFirebaseLegacyNotificationBody: () => createFirebaseLegacyNotificationBody,
  createFirebaseV1NotificationBody: () => createFirebaseV1NotificationBody,
  createWindowsBadgeNotificationBody: () => createWindowsBadgeNotificationBody
});
module.exports = __toCommonJS(notificationBodyBuilder_exports);
var import_core_xml = require("@azure/core-xml");
function createAppleNotificationBody(nativeMessage) {
  return JSON.stringify(nativeMessage);
}
function createFirebaseLegacyNotificationBody(nativeMessage) {
  return JSON.stringify(nativeMessage);
}
function createFirebaseV1NotificationBody(nativeMessage) {
  return JSON.stringify(nativeMessage);
}
function createAdmNotificationBody(nativeMessage) {
  return JSON.stringify(nativeMessage);
}
function createBaiduNotificationBody(nativeMessage) {
  return JSON.stringify(nativeMessage);
}
function createWindowsBadgeNotificationBody(nativeMessage) {
  const badge = {
    $: { value: nativeMessage.value }
  };
  return (0, import_core_xml.stringifyXML)(badge, { rootName: "badge" });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createAdmNotificationBody,
  createAppleNotificationBody,
  createBaiduNotificationBody,
  createFirebaseLegacyNotificationBody,
  createFirebaseV1NotificationBody,
  createWindowsBadgeNotificationBody
});
