// @azure/notification-hubs/dist/commonjs/models/notification.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notification_exports = {};
__export(notification_exports, {
  createAdmNotification: () => createAdmNotification,
  createAppleNotification: () => createAppleNotification,
  createBaiduNotification: () => createBaiduNotification,
  createBrowserNotification: () => createBrowserNotification,
  createFcmLegacyNotification: () => createFcmLegacyNotification,
  createFcmV1Notification: () => createFcmV1Notification,
  createTemplateNotification: () => createTemplateNotification,
  createWindowsBadgeNotification: () => createWindowsBadgeNotification,
  createWindowsNotification: () => createWindowsNotification,
  createWindowsRawNotification: () => createWindowsRawNotification,
  createWindowsTileNotification: () => createWindowsTileNotification,
  createWindowsToastNotification: () => createWindowsToastNotification,
  createXiaomiNotification: () => createXiaomiNotification
});
module.exports = __toCommonJS(notification_exports);
var Constants = __toESM(require("../utils/constants.js"));
function isString(value) {
  return typeof value === "string" || value instanceof String;
}
function createAppleNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "apple",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createAdmNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "adm",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createBaiduNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "baidu",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createBrowserNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "browser",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createFcmLegacyNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "gcm",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createFcmV1Notification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "fcmv1",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createXiaomiNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "xiaomi",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createTemplateNotification(notification) {
  const body = isString(notification.body) ? notification.body : JSON.stringify(notification.body);
  return {
    ...notification,
    body,
    platform: "template",
    contentType: Constants.JSON_CONTENT_TYPE
  };
}
function createWindowsNotification(notification) {
  if (notification?.headers && notification.headers["X-WNS-Type"]) {
    const wnsType = notification.headers["X-WNS-Type"];
    switch (wnsType) {
      case Constants.WNS_TOAST:
        return createWindowsToastNotification(notification);
      case Constants.WNS_TITLE:
        return createWindowsTileNotification(notification);
      case Constants.WNS_BADGE:
        return createWindowsBadgeNotification(notification);
      case Constants.WNS_RAW:
        return createWindowsRawNotification(notification);
      default:
        throw new Error(`Invalid WNS type: ${wnsType}`);
    }
  } else {
    throw new Error(`Missing WNS type in headers`);
  }
}
function createWindowsBadgeNotification(notification) {
  const result = {
    ...notification,
    platform: "windows",
    contentType: Constants.XML_CONTENT_TYPE
  };
  if (!result.headers) {
    result.headers = {};
  }
  if (!result.headers[Constants.WNS_TYPE_NAME]) {
    result.headers[Constants.WNS_TYPE_NAME] = Constants.WNS_BADGE;
  }
  return result;
}
function createWindowsTileNotification(notification) {
  const result = {
    ...notification,
    platform: "windows",
    contentType: Constants.XML_CONTENT_TYPE
  };
  if (!result.headers) {
    result.headers = {};
  }
  if (!result.headers[Constants.WNS_TYPE_NAME]) {
    result.headers[Constants.WNS_TYPE_NAME] = Constants.WNS_TITLE;
  }
  return result;
}
function createWindowsToastNotification(notification) {
  const result = {
    ...notification,
    platform: "windows",
    contentType: Constants.XML_CONTENT_TYPE
  };
  if (!result.headers) {
    result.headers = {};
  }
  if (!result.headers[Constants.WNS_TYPE_NAME]) {
    result.headers[Constants.WNS_TYPE_NAME] = Constants.WNS_TOAST;
  }
  return result;
}
function createWindowsRawNotification(notification) {
  const result = {
    ...notification,
    platform: "windows",
    contentType: Constants.STREAM_CONTENT_TYPE
  };
  if (!result.headers) {
    result.headers = {};
  }
  if (!result.headers[Constants.WNS_TYPE_NAME]) {
    result.headers[Constants.WNS_TYPE_NAME] = Constants.WNS_RAW;
  }
  return result;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createAdmNotification,
  createAppleNotification,
  createBaiduNotification,
  createBrowserNotification,
  createFcmLegacyNotification,
  createFcmV1Notification,
  createTemplateNotification,
  createWindowsBadgeNotification,
  createWindowsNotification,
  createWindowsRawNotification,
  createWindowsTileNotification,
  createWindowsToastNotification,
  createXiaomiNotification
});
