// @azure/notification-hubs/dist/commonjs/models/registration.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var registration_exports = {};
__export(registration_exports, {
  createAdmRegistrationDescription: () => createAdmRegistrationDescription,
  createAdmTemplateRegistrationDescription: () => createAdmTemplateRegistrationDescription,
  createAppleRegistrationDescription: () => createAppleRegistrationDescription,
  createAppleTemplateRegistrationDescription: () => createAppleTemplateRegistrationDescription,
  createBaiduRegistrationDescription: () => createBaiduRegistrationDescription,
  createBaiduTemplateRegistrationDescription: () => createBaiduTemplateRegistrationDescription,
  createBrowserRegistrationDescription: () => createBrowserRegistrationDescription,
  createBrowserTemplateRegistrationDescription: () => createBrowserTemplateRegistrationDescription,
  createFcmLegacyRegistrationDescription: () => createFcmLegacyRegistrationDescription,
  createFcmLegacyTemplateRegistrationDescription: () => createFcmLegacyTemplateRegistrationDescription,
  createFcmV1RegistrationDescription: () => createFcmV1RegistrationDescription,
  createFcmV1TemplateRegistrationDescription: () => createFcmV1TemplateRegistrationDescription,
  createWindowsRegistrationDescription: () => createWindowsRegistrationDescription,
  createWindowsTemplateRegistrationDescription: () => createWindowsTemplateRegistrationDescription,
  createXiaomiRegistrationDescription: () => createXiaomiRegistrationDescription,
  createXiaomiTemplateRegistrationDescription: () => createXiaomiTemplateRegistrationDescription
});
module.exports = __toCommonJS(registration_exports);
function createAdmRegistrationDescription(description) {
  return {
    ...description,
    kind: "Adm"
  };
}
function createAdmTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "AdmTemplate"
  };
}
function createAppleRegistrationDescription(description) {
  return {
    ...description,
    kind: "Apple"
  };
}
function createAppleTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "AppleTemplate"
  };
}
function createBaiduRegistrationDescription(description) {
  return {
    ...description,
    kind: "Baidu"
  };
}
function createBaiduTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "BaiduTemplate"
  };
}
function createBrowserRegistrationDescription(description) {
  return {
    ...description,
    kind: "Browser"
  };
}
function createBrowserTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "BrowserTemplate"
  };
}
function createFcmLegacyRegistrationDescription(description) {
  return {
    ...description,
    kind: "Gcm"
  };
}
function createFcmLegacyTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "GcmTemplate"
  };
}
function createFcmV1RegistrationDescription(description) {
  return {
    ...description,
    kind: "FcmV1"
  };
}
function createFcmV1TemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "FcmV1Template"
  };
}
function createWindowsRegistrationDescription(description) {
  return {
    ...description,
    kind: "Windows"
  };
}
function createWindowsTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "WindowsTemplate"
  };
}
function createXiaomiRegistrationDescription(description) {
  return {
    ...description,
    kind: "Xiaomi"
  };
}
function createXiaomiTemplateRegistrationDescription(description) {
  return {
    ...description,
    kind: "XiaomiTemplate"
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createAdmRegistrationDescription,
  createAdmTemplateRegistrationDescription,
  createAppleRegistrationDescription,
  createAppleTemplateRegistrationDescription,
  createBaiduRegistrationDescription,
  createBaiduTemplateRegistrationDescription,
  createBrowserRegistrationDescription,
  createBrowserTemplateRegistrationDescription,
  createFcmLegacyRegistrationDescription,
  createFcmLegacyTemplateRegistrationDescription,
  createFcmV1RegistrationDescription,
  createFcmV1TemplateRegistrationDescription,
  createWindowsRegistrationDescription,
  createWindowsTemplateRegistrationDescription,
  createXiaomiRegistrationDescription,
  createXiaomiTemplateRegistrationDescription
});
