// @azure/notification-hubs/dist/commonjs/serializers/notificationHubJobSerializer.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notificationHubJobSerializer_exports = {};
__export(notificationHubJobSerializer_exports, {
  parseNotificationHubJobEntry: () => parseNotificationHubJobEntry,
  parseNotificationHubJobFeed: () => parseNotificationHubJobFeed,
  serializeNotificationHubJobEntry: () => serializeNotificationHubJobEntry
});
module.exports = __toCommonJS(notificationHubJobSerializer_exports);
var import_utils = require("../utils/utils.js");
var import_core_xml = require("@azure/core-xml");
var import_xmlUtils = require("../utils/xmlUtils.js");
function serializeNotificationHubJobEntry(entry) {
  const job = {
    Type: entry.type,
    OutputContainerUri: { __cdata: entry.outputContainerUrl },
    ImportFileUri: (0, import_utils.isDefined)(entry.importFileUrl) ? { __cdata: entry.importFileUrl } : void 0
  };
  const requestObject = (0, import_xmlUtils.serializeToAtomXmlRequest)("NotificationHubJob", job);
  return (0, import_core_xml.stringifyXML)(requestObject, { rootName: "entry", cdataPropName: "__cdata" });
}
async function parseNotificationHubJobEntry(bodyText) {
  const xml = await (0, import_core_xml.parseXML)(bodyText, { includeRoot: true });
  const content = xml.entry.content.NotificationHubJob;
  return createNotificationHubJob(content);
}
async function parseNotificationHubJobFeed(bodyText) {
  const xml = await (0, import_core_xml.parseXML)(bodyText, { includeRoot: true });
  const results = [];
  if (!(0, import_utils.isDefined)(xml.feed.entry)) {
    return results;
  }
  const entries = Array.isArray(xml.feed.entry) ? xml.feed.entry : [xml.feed.entry];
  for (const item of entries) {
    results.push(createNotificationHubJob(item.content.NotificationHubJob));
  }
  return results;
}
function createInputOutputProperties(content) {
  const props = {};
  const keyValues = content["d3p1:KeyValueOfstringstring"];
  const keyValueArray = Array.isArray(keyValues) ? keyValues : [keyValues];
  for (const item of keyValueArray) {
    props[item["d3p1:Key"]] = item["d3p1:Value"];
  }
  return props;
}
function createNotificationHubJob(content) {
  let outputProperties;
  if ((0, import_utils.isDefined)(content["OutputProperties"])) {
    outputProperties = createInputOutputProperties(content["OutputProperties"]);
  }
  let inputProperties;
  if ((0, import_utils.isDefined)(content["InputProperties"])) {
    inputProperties = createInputOutputProperties(content["InputProperties"]);
  }
  return {
    jobId: (0, import_utils.getStringOrUndefined)(content["JobId"])?.trim(),
    type: (0, import_utils.getString)(content["Type"], "type").trim(),
    status: (0, import_utils.getStringOrUndefined)(content["Status"])?.trim(),
    progress: (0, import_utils.getFloatOrUndefined)(content["Progress"]),
    outputContainerUrl: (0, import_utils.getString)(content["OutputContainerUri"], "outputContainerUrl").trim(),
    importFileUrl: (0, import_utils.getStringOrUndefined)(content["ImportFileUri"])?.trim(),
    failure: (0, import_utils.getStringOrUndefined)(content["Failure"])?.trim(),
    createdAt: (0, import_utils.getDateOrUndefined)(content["CreatedAt"]),
    updatedAt: (0, import_utils.getDateOrUndefined)(content["UpdatedAt"]),
    inputProperties,
    outputProperties
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  parseNotificationHubJobEntry,
  parseNotificationHubJobFeed,
  serializeNotificationHubJobEntry
});
