// @azure/notification-hubs/dist/commonjs/serializers/notificationDetailsSerializer.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notificationDetailsSerializer_exports = {};
__export(notificationDetailsSerializer_exports, {
  parseNotificationDetails: () => parseNotificationDetails
});
module.exports = __toCommonJS(notificationDetailsSerializer_exports);
var import_utils = require("../utils/utils.js");
var import_core_xml = require("@azure/core-xml");
async function parseNotificationDetails(bodyText) {
  const xml = await (0, import_core_xml.parseXML)(bodyText, {
    includeRoot: true
  });
  const notificationDetails = xml["NotificationDetails"];
  let apnsOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["ApnsOutcomeCounts"])) {
    apnsOutcomeCounts = parseOutcomeCounts(notificationDetails["ApnsOutcomeCounts"]["Outcome"]);
  }
  let admOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["AdmOutcomeCounts"])) {
    admOutcomeCounts = parseOutcomeCounts(notificationDetails["AdmOutcomeCounts"]["Outcome"]);
  }
  let baiduOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["BaiduOutcomeCounts"])) {
    baiduOutcomeCounts = parseOutcomeCounts(notificationDetails["BaiduOutcomeCounts"]["Outcome"]);
  }
  let browserOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["BrowserOutcomeCounts"])) {
    browserOutcomeCounts = parseOutcomeCounts(
      notificationDetails["BrowserOutcomeCounts"]["Outcome"]
    );
  }
  let fcmOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["GcmOutcomeCounts"])) {
    fcmOutcomeCounts = parseOutcomeCounts(notificationDetails["GcmOutcomeCounts"]["Outcome"]);
  }
  let fcmV1OutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["FcmV1OutcomeCounts"])) {
    fcmV1OutcomeCounts = parseOutcomeCounts(notificationDetails["FcmV1OutcomeCounts"]["Outcome"]);
  }
  let xiaomiOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["XiaomiOutcomeCounts"])) {
    xiaomiOutcomeCounts = parseOutcomeCounts(notificationDetails["XiaomiOutcomeCounts"]["Outcome"]);
  }
  let wnsOutcomeCounts;
  if ((0, import_utils.isDefined)(notificationDetails["WnsOutcomeCounts"])) {
    wnsOutcomeCounts = parseOutcomeCounts(notificationDetails["WnsOutcomeCounts"]["Outcome"]);
  }
  return {
    notificationId: (0, import_utils.getStringOrUndefined)(notificationDetails["NotificationId"])?.trim(),
    location: (0, import_utils.getStringOrUndefined)(notificationDetails["Location"])?.trim(),
    state: (0, import_utils.getStringOrUndefined)(notificationDetails["State"])?.trim(),
    enqueueTime: (0, import_utils.getDateOrUndefined)(notificationDetails["EnqueueTime"]),
    startTime: (0, import_utils.getDateOrUndefined)(notificationDetails["StartTime"]),
    endTime: (0, import_utils.getDateOrUndefined)(notificationDetails["EndTime"]),
    pnsErrorDetailsUrl: (0, import_utils.getStringOrUndefined)(notificationDetails["PnsErrorDetailsUri"])?.trim(),
    targetPlatforms: (0, import_utils.getStringOrUndefined)(notificationDetails["TargetPlatforms"])?.trim(),
    notificationBody: (0, import_utils.getStringOrUndefined)(notificationDetails["NotificationBody"])?.trim(),
    apnsOutcomeCounts,
    admOutcomeCounts,
    baiduOutcomeCounts,
    browserOutcomeCounts,
    fcmOutcomeCounts,
    fcmV1OutcomeCounts,
    xiaomiOutcomeCounts,
    wnsOutcomeCounts
  };
}
function parseOutcomeCounts(counts) {
  const items = Array.isArray(counts) ? counts : [counts];
  const results = [];
  for (const item of items) {
    results.push({ state: item["Name"], count: (0, import_utils.getInteger)(item["Count"], "Count") });
  }
  return results;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  parseNotificationDetails
});
