// @azure/notification-hubs/dist/commonjs/serializers/notificationOutcomeSerializer.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notificationOutcomeSerializer_exports = {};
__export(notificationOutcomeSerializer_exports, {
  parseNotificationOutcome: () => parseNotificationOutcome
});
module.exports = __toCommonJS(notificationOutcomeSerializer_exports);
var import_utils = require("../utils/utils.js");
var import_core_xml = require("@azure/core-xml");
async function parseNotificationOutcome(bodyText) {
  const xml = await (0, import_core_xml.parseXML)(bodyText, { includeRoot: true });
  const outcome = xml.NotificationOutcome;
  return {
    successCount: (0, import_utils.getInteger)(outcome.Success, "Success"),
    failureCount: (0, import_utils.getInteger)(outcome.Failure, "Failure"),
    results: parseRegistrationResults(outcome.Results.RegistrationResult),
    state: "DetailedStateAvailable"
  };
}
function parseRegistrationResults(results) {
  const registrationResults = [];
  if (!(0, import_utils.isDefined)(results)) {
    return registrationResults;
  }
  const resultsArray = Array.isArray(results) ? results : [results];
  for (const result of resultsArray) {
    registrationResults.push({
      applicationPlatform: (0, import_utils.getString)(result.ApplicationPlatform, "ApplicationPlatform").trim(),
      registrationId: (0, import_utils.getString)(result.RegistrationId, "RegistrationId").trim(),
      outcome: (0, import_utils.getString)(result.Outcome, "Outcome").trim(),
      pnsHandle: (0, import_utils.getString)(result.PnsHandle, "PnsHandle").trim()
    });
  }
  return registrationResults;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  parseNotificationOutcome
});
