// @azure/arm-policy/dist/commonjs/api/policyDefinitionVersions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyDefinitionVersions_exports = {};
__export(policyDefinitionVersions_exports, {
  $delete: () => import_operations.$delete,
  PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsCreateOrUpdateOptionalParams: () => import_options.PolicyDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsDeleteOptionalParams: () => import_options.PolicyDefinitionVersionsDeleteOptionalParams,
  PolicyDefinitionVersionsGetAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsGetBuiltInOptionalParams: () => import_options.PolicyDefinitionVersionsGetBuiltInOptionalParams,
  PolicyDefinitionVersionsGetOptionalParams: () => import_options.PolicyDefinitionVersionsGetOptionalParams,
  PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsListAllBuiltinsOptionalParams: () => import_options.PolicyDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicyDefinitionVersionsListAllOptionalParams: () => import_options.PolicyDefinitionVersionsListAllOptionalParams,
  PolicyDefinitionVersionsListBuiltInOptionalParams: () => import_options.PolicyDefinitionVersionsListBuiltInOptionalParams,
  PolicyDefinitionVersionsListByManagementGroupOptionalParams: () => import_options.PolicyDefinitionVersionsListByManagementGroupOptionalParams,
  PolicyDefinitionVersionsListOptionalParams: () => import_options.PolicyDefinitionVersionsListOptionalParams,
  createOrUpdate: () => import_operations.createOrUpdate,
  createOrUpdateAtManagementGroup: () => import_operations.createOrUpdateAtManagementGroup,
  deleteAtManagementGroup: () => import_operations.deleteAtManagementGroup,
  get: () => import_operations.get,
  getAtManagementGroup: () => import_operations.getAtManagementGroup,
  getBuiltIn: () => import_operations.getBuiltIn,
  list: () => import_operations.list,
  listAll: () => import_operations.listAll,
  listAllAtManagementGroup: () => import_operations.listAllAtManagementGroup,
  listAllBuiltins: () => import_operations.listAllBuiltins,
  listBuiltIn: () => import_operations.listBuiltIn,
  listByManagementGroup: () => import_operations.listByManagementGroup
});
module.exports = __toCommonJS(policyDefinitionVersions_exports);
var import_operations = require("./operations.js");
var import_options = require("./options.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsDeleteOptionalParams,
  PolicyDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsGetBuiltInOptionalParams,
  PolicyDefinitionVersionsGetOptionalParams,
  PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicyDefinitionVersionsListAllOptionalParams,
  PolicyDefinitionVersionsListBuiltInOptionalParams,
  PolicyDefinitionVersionsListByManagementGroupOptionalParams,
  PolicyDefinitionVersionsListOptionalParams,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listAll,
  listAllAtManagementGroup,
  listAllBuiltins,
  listBuiltIn,
  listByManagementGroup
});
