// @azure/arm-policy/dist/commonjs/api/policyDefinitionVersions/operations.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var operations_exports = {};
__export(operations_exports, {
  $delete: () => $delete,
  _$deleteDeserialize: () => _$deleteDeserialize,
  _$deleteSend: () => _$deleteSend,
  _createOrUpdateAtManagementGroupDeserialize: () => _createOrUpdateAtManagementGroupDeserialize,
  _createOrUpdateAtManagementGroupSend: () => _createOrUpdateAtManagementGroupSend,
  _createOrUpdateDeserialize: () => _createOrUpdateDeserialize,
  _createOrUpdateSend: () => _createOrUpdateSend,
  _deleteAtManagementGroupDeserialize: () => _deleteAtManagementGroupDeserialize,
  _deleteAtManagementGroupSend: () => _deleteAtManagementGroupSend,
  _getAtManagementGroupDeserialize: () => _getAtManagementGroupDeserialize,
  _getAtManagementGroupSend: () => _getAtManagementGroupSend,
  _getBuiltInDeserialize: () => _getBuiltInDeserialize,
  _getBuiltInSend: () => _getBuiltInSend,
  _getDeserialize: () => _getDeserialize,
  _getSend: () => _getSend,
  _listAllAtManagementGroupDeserialize: () => _listAllAtManagementGroupDeserialize,
  _listAllAtManagementGroupSend: () => _listAllAtManagementGroupSend,
  _listAllBuiltinsDeserialize: () => _listAllBuiltinsDeserialize,
  _listAllBuiltinsSend: () => _listAllBuiltinsSend,
  _listAllDeserialize: () => _listAllDeserialize,
  _listAllSend: () => _listAllSend,
  _listBuiltInDeserialize: () => _listBuiltInDeserialize,
  _listBuiltInSend: () => _listBuiltInSend,
  _listByManagementGroupDeserialize: () => _listByManagementGroupDeserialize,
  _listByManagementGroupSend: () => _listByManagementGroupSend,
  _listDeserialize: () => _listDeserialize,
  _listSend: () => _listSend,
  createOrUpdate: () => createOrUpdate,
  createOrUpdateAtManagementGroup: () => createOrUpdateAtManagementGroup,
  deleteAtManagementGroup: () => deleteAtManagementGroup,
  get: () => get,
  getAtManagementGroup: () => getAtManagementGroup,
  getBuiltIn: () => getBuiltIn,
  list: () => list,
  listAll: () => listAll,
  listAllAtManagementGroup: () => listAllAtManagementGroup,
  listAllBuiltins: () => listAllBuiltins,
  listBuiltIn: () => listBuiltIn,
  listByManagementGroup: () => listByManagementGroup
});
module.exports = __toCommonJS(operations_exports);
var import_models = require("../../models/models.js");
var import_pagingHelpers = require("../../static-helpers/pagingHelpers.js");
var import_urlTemplate = require("../../static-helpers/urlTemplate.js");
var import_core_client = require("@azure-rest/core-client");
function _listAllSend(context, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).post({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listAllDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
async function listAll(context, options = { requestOptions: {} }) {
  const result = await _listAllSend(context, options);
  return _listAllDeserialize(result);
}
function _listAllAtManagementGroupSend(context, managementGroupName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}",
    {
      managementGroupName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).post({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listAllAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
async function listAllAtManagementGroup(context, managementGroupName, options = { requestOptions: {} }) {
  const result = await _listAllAtManagementGroupSend(context, managementGroupName, options);
  return _listAllAtManagementGroupDeserialize(result);
}
function _listAllBuiltinsSend(context, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}",
    {
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).post({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listAllBuiltinsDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
async function listAllBuiltins(context, options = { requestOptions: {} }) {
  const result = await _listAllBuiltinsSend(context, options);
  return _listAllBuiltinsDeserialize(result);
}
function _listByManagementGroupSend(context, managementGroupName, policyDefinitionName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}",
    {
      managementGroupName,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listByManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
function listByManagementGroup(context, managementGroupName, policyDefinitionName, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listByManagementGroupSend(context, managementGroupName, policyDefinitionName, options),
    _listByManagementGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _deleteAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({ ...(0, import_core_client.operationOptionsToRequestParameters)(options) });
}
async function _deleteAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return;
}
async function deleteAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const result = await _deleteAtManagementGroupSend(
    context,
    managementGroupName,
    policyDefinitionName,
    policyDefinitionVersion,
    options
  );
  return _deleteAtManagementGroupDeserialize(result);
}
function _createOrUpdateAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options = {
  requestOptions: {}
}) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: (0, import_models.policyDefinitionVersionSerializer)(parameters)
  });
}
async function _createOrUpdateAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200", "201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyDefinitionVersionDeserializer)(result.body);
}
async function createOrUpdateAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options = {
  requestOptions: {}
}) {
  const result = await _createOrUpdateAtManagementGroupSend(
    context,
    managementGroupName,
    policyDefinitionName,
    policyDefinitionVersion,
    parameters,
    options
  );
  return _createOrUpdateAtManagementGroupDeserialize(result);
}
function _getAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyDefinitionVersionDeserializer)(result.body);
}
async function getAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const result = await _getAtManagementGroupSend(
    context,
    managementGroupName,
    policyDefinitionName,
    policyDefinitionVersion,
    options
  );
  return _getAtManagementGroupDeserialize(result);
}
function _listBuiltInSend(context, policyDefinitionName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}",
    {
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listBuiltInDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
function listBuiltIn(context, policyDefinitionName, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listBuiltInSend(context, policyDefinitionName, options),
    _listBuiltInDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _getBuiltInSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getBuiltInDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyDefinitionVersionDeserializer)(result.body);
}
async function getBuiltIn(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const result = await _getBuiltInSend(
    context,
    policyDefinitionName,
    policyDefinitionVersion,
    options
  );
  return _getBuiltInDeserialize(result);
}
function _listSend(context, policyDefinitionName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyDefinitionVersionListResultDeserializer)(result.body);
}
function list(context, policyDefinitionName, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listSend(context, policyDefinitionName, options),
    _listDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _$deleteSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({ ...(0, import_core_client.operationOptionsToRequestParameters)(options) });
}
async function _$deleteDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return;
}
async function $delete(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const result = await _$deleteSend(
    context,
    policyDefinitionName,
    policyDefinitionVersion,
    options
  );
  return _$deleteDeserialize(result);
}
function _createOrUpdateSend(context, policyDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: (0, import_models.policyDefinitionVersionSerializer)(parameters)
  });
}
async function _createOrUpdateDeserialize(result) {
  const expectedStatuses = ["200", "201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyDefinitionVersionDeserializer)(result.body);
}
async function createOrUpdate(context, policyDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
  const result = await _createOrUpdateSend(
    context,
    policyDefinitionName,
    policyDefinitionVersion,
    parameters,
    options
  );
  return _createOrUpdateDeserialize(result);
}
function _getSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      policyDefinitionName,
      policyDefinitionVersion,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyDefinitionVersionDeserializer)(result.body);
}
async function get(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
  const result = await _getSend(context, policyDefinitionName, policyDefinitionVersion, options);
  return _getDeserialize(result);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  _$deleteDeserialize,
  _$deleteSend,
  _createOrUpdateAtManagementGroupDeserialize,
  _createOrUpdateAtManagementGroupSend,
  _createOrUpdateDeserialize,
  _createOrUpdateSend,
  _deleteAtManagementGroupDeserialize,
  _deleteAtManagementGroupSend,
  _getAtManagementGroupDeserialize,
  _getAtManagementGroupSend,
  _getBuiltInDeserialize,
  _getBuiltInSend,
  _getDeserialize,
  _getSend,
  _listAllAtManagementGroupDeserialize,
  _listAllAtManagementGroupSend,
  _listAllBuiltinsDeserialize,
  _listAllBuiltinsSend,
  _listAllDeserialize,
  _listAllSend,
  _listBuiltInDeserialize,
  _listBuiltInSend,
  _listByManagementGroupDeserialize,
  _listByManagementGroupSend,
  _listDeserialize,
  _listSend,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listAll,
  listAllAtManagementGroup,
  listAllBuiltins,
  listBuiltIn,
  listByManagementGroup
});
