// @azure/arm-policy/dist/commonjs/classic/policySetDefinitionVersions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policySetDefinitionVersions_exports = {};
__export(policySetDefinitionVersions_exports, {
  _getPolicySetDefinitionVersionsOperations: () => _getPolicySetDefinitionVersionsOperations
});
module.exports = __toCommonJS(policySetDefinitionVersions_exports);
var import_operations = require("../../api/policySetDefinitionVersions/operations.js");
function _getPolicySetDefinitionVersions(context) {
  return {
    listAll: (options) => (0, import_operations.listAll)(context, options),
    listAllAtManagementGroup: (managementGroupName, options) => (0, import_operations.listAllAtManagementGroup)(context, managementGroupName, options),
    listAllBuiltins: (options) => (0, import_operations.listAllBuiltins)(context, options),
    listByManagementGroup: (managementGroupName, policySetDefinitionName, options) => (0, import_operations.listByManagementGroup)(context, managementGroupName, policySetDefinitionName, options),
    deleteAtManagementGroup: (managementGroupName, policySetDefinitionName, policyDefinitionVersion, options) => (0, import_operations.deleteAtManagementGroup)(
      context,
      managementGroupName,
      policySetDefinitionName,
      policyDefinitionVersion,
      options
    ),
    createOrUpdateAtManagementGroup: (managementGroupName, policySetDefinitionName, policyDefinitionVersion, parameters, options) => (0, import_operations.createOrUpdateAtManagementGroup)(
      context,
      managementGroupName,
      policySetDefinitionName,
      policyDefinitionVersion,
      parameters,
      options
    ),
    getAtManagementGroup: (managementGroupName, policySetDefinitionName, policyDefinitionVersion, options) => (0, import_operations.getAtManagementGroup)(
      context,
      managementGroupName,
      policySetDefinitionName,
      policyDefinitionVersion,
      options
    ),
    listBuiltIn: (policySetDefinitionName, options) => (0, import_operations.listBuiltIn)(context, policySetDefinitionName, options),
    getBuiltIn: (policySetDefinitionName, policyDefinitionVersion, options) => (0, import_operations.getBuiltIn)(context, policySetDefinitionName, policyDefinitionVersion, options),
    list: (policySetDefinitionName, options) => (0, import_operations.list)(context, policySetDefinitionName, options),
    delete: (policySetDefinitionName, policyDefinitionVersion, options) => (0, import_operations.$delete)(context, policySetDefinitionName, policyDefinitionVersion, options),
    createOrUpdate: (policySetDefinitionName, policyDefinitionVersion, parameters, options) => (0, import_operations.createOrUpdate)(
      context,
      policySetDefinitionName,
      policyDefinitionVersion,
      parameters,
      options
    ),
    get: (policySetDefinitionName, policyDefinitionVersion, options) => (0, import_operations.get)(context, policySetDefinitionName, policyDefinitionVersion, options)
  };
}
function _getPolicySetDefinitionVersionsOperations(context) {
  return {
    ..._getPolicySetDefinitionVersions(context)
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  _getPolicySetDefinitionVersionsOperations
});
