// @azure/arm-policy/dist/commonjs/classic/policyAssignments/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyAssignments_exports = {};
__export(policyAssignments_exports, {
  _getPolicyAssignmentsOperations: () => _getPolicyAssignmentsOperations
});
module.exports = __toCommonJS(policyAssignments_exports);
var import_operations = require("../../api/policyAssignments/operations.js");
function _getPolicyAssignments(context) {
  return {
    listForResource: (resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, options) => (0, import_operations.listForResource)(
      context,
      resourceGroupName,
      resourceProviderNamespace,
      parentResourcePath,
      resourceType,
      resourceName,
      options
    ),
    list: (options) => (0, import_operations.list)(context, options),
    listForManagementGroup: (managementGroupId, options) => (0, import_operations.listForManagementGroup)(context, managementGroupId, options),
    listForResourceGroup: (resourceGroupName, options) => (0, import_operations.listForResourceGroup)(context, resourceGroupName, options),
    delete: (scope, policyAssignmentName, options) => (0, import_operations.$delete)(context, scope, policyAssignmentName, options),
    update: (scope, policyAssignmentName, parameters, options) => (0, import_operations.update)(context, scope, policyAssignmentName, parameters, options),
    create: (scope, policyAssignmentName, parameters, options) => (0, import_operations.create)(context, scope, policyAssignmentName, parameters, options),
    get: (scope, policyAssignmentName, options) => (0, import_operations.get)(context, scope, policyAssignmentName, options)
  };
}
function _getPolicyAssignmentsOperations(context) {
  return {
    ..._getPolicyAssignments(context)
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  _getPolicyAssignmentsOperations
});
