// @azure/arm-policy/dist/esm/api/policyTokens/operations.js
import {
  errorResponseDeserializer,
  policyTokenRequestSerializer,
  policyTokenResponseDeserializer
} from "../../models/models.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
function _acquireAtManagementGroupSend(context, managementGroupName, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/acquirePolicyToken{?api%2Dversion}",
    {
      managementGroupName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).post({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyTokenRequestSerializer(parameters)
  });
}
async function _acquireAtManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyTokenResponseDeserializer(result.body);
}
async function acquireAtManagementGroup(context, managementGroupName, parameters, options = { requestOptions: {} }) {
  const result = await _acquireAtManagementGroupSend(
    context,
    managementGroupName,
    parameters,
    options
  );
  return _acquireAtManagementGroupDeserialize(result);
}
function _acquireSend(context, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/acquirePolicyToken{?api%2Dversion}",
    {
      subscriptionId: context.subscriptionId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).post({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyTokenRequestSerializer(parameters)
  });
}
async function _acquireDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyTokenResponseDeserializer(result.body);
}
async function acquire(context, parameters, options = { requestOptions: {} }) {
  const result = await _acquireSend(context, parameters, options);
  return _acquireDeserialize(result);
}
export {
  _acquireAtManagementGroupDeserialize,
  _acquireAtManagementGroupSend,
  _acquireDeserialize,
  _acquireSend,
  acquire,
  acquireAtManagementGroup
};
