// @azure/arm-policy/dist/esm/api/policyAssignments/operations.js
import {
  policyAssignmentSerializer,
  policyAssignmentDeserializer,
  errorResponseDeserializer,
  policyAssignmentUpdateSerializer,
  _policyAssignmentListResultDeserializer
} from "../../models/models.js";
import { buildPagedAsyncIterator } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
function _listForResourceSend(context, resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{+parentResourcePath}/{+resourceType}/{resourceName}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      resourceGroupName,
      resourceProviderNamespace,
      parentResourcePath,
      resourceType,
      resourceName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForResourceDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyAssignmentListResultDeserializer(result.body);
}
function listForResource(context, resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listForResourceSend(
      context,
      resourceGroupName,
      resourceProviderNamespace,
      parentResourcePath,
      resourceType,
      resourceName,
      options
    ),
    _listForResourceDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listSend(context, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyAssignmentListResultDeserializer(result.body);
}
function list(context, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listSend(context, options),
    _listDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listForManagementGroupSend(context, managementGroupId, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      managementGroupId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyAssignmentListResultDeserializer(result.body);
}
function listForManagementGroup(context, managementGroupId, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listForManagementGroupSend(context, managementGroupId, options),
    _listForManagementGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listForResourceGroupSend(context, resourceGroupName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      resourceGroupName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForResourceGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return _policyAssignmentListResultDeserializer(result.body);
}
function listForResourceGroup(context, resourceGroupName, options = { requestOptions: {} }) {
  return buildPagedAsyncIterator(
    context,
    () => _listForResourceGroupSend(context, resourceGroupName, options),
    _listForResourceGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _$deleteSend(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _$deleteDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyAssignmentDeserializer(result.body);
}
async function $delete(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const result = await _$deleteSend(context, scope, policyAssignmentName, options);
  return _$deleteDeserialize(result);
}
function _updateSend(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).patch({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyAssignmentUpdateSerializer(parameters)
  });
}
async function _updateDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyAssignmentDeserializer(result.body);
}
async function update(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const result = await _updateSend(context, scope, policyAssignmentName, parameters, options);
  return _updateDeserialize(result);
}
function _createSend(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...operationOptionsToRequestParameters(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: policyAssignmentSerializer(parameters)
  });
}
async function _createDeserialize(result) {
  const expectedStatuses = ["201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyAssignmentDeserializer(result.body);
}
async function create(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const result = await _createSend(context, scope, policyAssignmentName, parameters, options);
  return _createDeserialize(result);
}
function _getSend(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const path = expandUrlTemplate(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion,%24expand}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24expand": options?.expand
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...operationOptionsToRequestParameters(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = createRestError(result);
    error.details = errorResponseDeserializer(result.body);
    throw error;
  }
  return policyAssignmentDeserializer(result.body);
}
async function get(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const result = await _getSend(context, scope, policyAssignmentName, options);
  return _getDeserialize(result);
}
export {
  $delete,
  _$deleteDeserialize,
  _$deleteSend,
  _createDeserialize,
  _createSend,
  _getDeserialize,
  _getSend,
  _listDeserialize,
  _listForManagementGroupDeserialize,
  _listForManagementGroupSend,
  _listForResourceDeserialize,
  _listForResourceGroupDeserialize,
  _listForResourceGroupSend,
  _listForResourceSend,
  _listSend,
  _updateDeserialize,
  _updateSend,
  create,
  get,
  list,
  listForManagementGroup,
  listForResource,
  listForResourceGroup,
  update
};
