// @azure/notification-hubs/dist/commonjs/auth/connectionStringUtils.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var connectionStringUtils_exports = {};
__export(connectionStringUtils_exports, {
  createTokenCredentialFromConnection: () => createTokenCredentialFromConnection,
  parseNotificationHubsConnectionString: () => parseNotificationHubsConnectionString
});
module.exports = __toCommonJS(connectionStringUtils_exports);
var import_sasTokenCredential = require("./sasTokenCredential.js");
function parseConnectionString(connectionString) {
  const output = {};
  const parts = connectionString.trim().split(";");
  for (let part of parts) {
    part = part.trim();
    if (part === "") {
      continue;
    }
    const splitIndex = part.indexOf("=");
    if (splitIndex === -1) {
      throw new Error(
        "Connection string malformed: each part of the connection string must have an `=` assignment."
      );
    }
    const key = part.substring(0, splitIndex).trim();
    if (key === "") {
      throw new Error("Connection string malformed: missing key for assignment");
    }
    const value = part.substring(splitIndex + 1).trim();
    output[key] = value;
  }
  return output;
}
function createTokenCredentialFromConnection(sharedAccessKey, sharedAccessKeyName) {
  return new import_sasTokenCredential.SasTokenCredential({ sharedAccessKey, sharedAccessKeyName });
}
function parseNotificationHubsConnectionString(connectionString) {
  const parsedResult = parseConnectionString(connectionString);
  if (!parsedResult.Endpoint) {
    throw new Error("Connection string should have an Endpoint key.");
  }
  if (parsedResult.SharedAccessKey && !parsedResult.SharedAccessKeyName) {
    throw new Error("Connection string with SharedAccessKey should have SharedAccessKeyName.");
  } else if (!parsedResult.SharedAccessKey && parsedResult.SharedAccessKeyName) {
    throw new Error(
      "Connection string with SharedAccessKeyName should have SharedAccessKey as well."
    );
  }
  const output = {
    endpoint: parsedResult.Endpoint,
    sharedAccessKey: parsedResult.SharedAccessKey,
    sharedAccessKeyName: parsedResult.SharedAccessKeyName
  };
  return output;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createTokenCredentialFromConnection,
  parseNotificationHubsConnectionString
});
