// @azure/arm-policy/dist/browser/classic/policyDefinitions/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { listByManagementGroup, deleteAtManagementGroup, createOrUpdateAtManagementGroup, getAtManagementGroup, listBuiltIn, getBuiltIn, list, $delete, createOrUpdate, get, } from "../../api/policyDefinitions/operations.js";
function _getPolicyDefinitions(context) {
    return {
        listByManagementGroup: (managementGroupId, options) => listByManagementGroup(context, managementGroupId, options),
        deleteAtManagementGroup: (managementGroupId, policyDefinitionName, options) => deleteAtManagementGroup(context, managementGroupId, policyDefinitionName, options),
        createOrUpdateAtManagementGroup: (managementGroupId, policyDefinitionName, parameters, options) => createOrUpdateAtManagementGroup(context, managementGroupId, policyDefinitionName, parameters, options),
        getAtManagementGroup: (managementGroupId, policyDefinitionName, options) => getAtManagementGroup(context, managementGroupId, policyDefinitionName, options),
        listBuiltIn: (options) => listBuiltIn(context, options),
        getBuiltIn: (policyDefinitionName, options) => getBuiltIn(context, policyDefinitionName, options),
        list: (options) => list(context, options),
        delete: (policyDefinitionName, options) => $delete(context, policyDefinitionName, options),
        createOrUpdate: (policyDefinitionName, parameters, options) => createOrUpdate(context, policyDefinitionName, parameters, options),
        get: (policyDefinitionName, options) => get(context, policyDefinitionName, options),
    };
}
export function _getPolicyDefinitionsOperations(context) {
    return {
        ..._getPolicyDefinitions(context),
    };
}
