// @azure/notification-hubs/dist/commonjs/api/internal/_client.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var client_exports = {};
__export(client_exports, {
  createRequest: () => createRequest,
  parseNotificationResponse: () => parseNotificationResponse,
  parseNotificationSendResponse: () => parseNotificationSendResponse,
  sendRequest: () => sendRequest
});
module.exports = __toCommonJS(client_exports);
var import_core_rest_pipeline = require("@azure/core-rest-pipeline");
var import_utils = require("../../utils/utils.js");
var import_notificationOutcomeSerializer = require("../../serializers/notificationOutcomeSerializer.js");
var import_xmlUtils = require("../../utils/xmlUtils.js");
function createRequest(endpoint, method, headers, options) {
  return (0, import_core_rest_pipeline.createPipelineRequest)({
    ...options.tracingOptions,
    ...options.requestOptions,
    url: endpoint.toString(),
    abortSignal: options.abortSignal,
    method,
    headers
  });
}
function parseNotificationResponse(response) {
  const correlationId = response.headers.get("x-ms-correlation-request-id");
  const trackingId = response.headers.get("TrackingId");
  const location = response.headers.get("Location");
  return {
    correlationId,
    trackingId,
    location
  };
}
async function parseNotificationSendResponse(response) {
  const result = parseNotificationResponse(response);
  let notificationId;
  if (result.location) {
    const locationUrl = new URL(result.location);
    notificationId = locationUrl.pathname.split("/")[3];
  }
  const requestUrl = new URL(response.request.url);
  const isTestSend = requestUrl.searchParams.has("test");
  const isDirectSend = requestUrl.searchParams.has("direct");
  const responseBody = response.bodyAsText;
  if (isTestSend && !isDirectSend && (0, import_utils.isDefined)(responseBody)) {
    const outcome = await (0, import_notificationOutcomeSerializer.parseNotificationOutcome)(responseBody);
    return {
      ...result,
      ...outcome,
      notificationId
    };
  } else {
    return createDefaultResponse(result, notificationId);
  }
}
function createDefaultResponse(response, notificationId) {
  return {
    ...response,
    notificationId,
    successCount: 0,
    failureCount: 0,
    results: [],
    state: "Enqueued"
  };
}
async function sendRequest(context, request, successStatusCode) {
  const statuses = Array.isArray(successStatusCode) ? successStatusCode : [successStatusCode];
  const response = await context.sendRequest(request);
  if (!statuses.some((statusCode) => statusCode === response.status)) {
    const responseBody = response.bodyAsText;
    let details;
    if ((0, import_utils.isDefined)(responseBody)) {
      details = await (0, import_xmlUtils.parseXMLError)(responseBody);
    }
    let errorMessage;
    if ((0, import_utils.isDefined)(details)) {
      errorMessage = `operations failed with: ${details}`;
    } else {
      errorMessage = `operation failed with status ${response.status}`;
    }
    throw new import_core_rest_pipeline.RestError(errorMessage, {
      statusCode: response.status,
      response
    });
  }
  return response;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createRequest,
  parseNotificationResponse,
  parseNotificationSendResponse,
  sendRequest
});
