// @azure/arm-policy/dist/esm/policyClient.js
import { createPolicy } from "./api/index.js";
import { _getPolicyAssignmentsOperations } from "./classic/policyAssignments/index.js";
import { _getPolicyDefinitionVersionsOperations } from "./classic/policyDefinitionVersions/index.js";
import { _getPolicyDefinitionsOperations } from "./classic/policyDefinitions/index.js";
import { _getPolicySetDefinitionVersionsOperations } from "./classic/policySetDefinitionVersions/index.js";
import { _getPolicySetDefinitionsOperations } from "./classic/policySetDefinitions/index.js";
import { _getPolicyTokensOperations } from "./classic/policyTokens/index.js";
import { PolicyClientOptionalParams } from "./api/policyContext.js";
class PolicyClient {
  _client;
  /** The pipeline used by this client to make requests */
  pipeline;
  /** To manage and control access to your resources, you can define customized policies and assign them at a scope. */
  constructor(credential, subscriptionIdOrOptions, options) {
    let subscriptionId;
    if (typeof subscriptionIdOrOptions === "string") {
      subscriptionId = subscriptionIdOrOptions;
    } else if (typeof subscriptionIdOrOptions === "object") {
      options = subscriptionIdOrOptions;
    }
    options = options ?? {};
    const prefixFromOptions = options?.userAgentOptions?.userAgentPrefix;
    const userAgentPrefix = prefixFromOptions ? `${prefixFromOptions} azsdk-js-client` : `azsdk-js-client`;
    this._client = createPolicy(credential, subscriptionId ?? "", {
      ...options,
      userAgentOptions: { userAgentPrefix }
    });
    this.pipeline = this._client.pipeline;
    this.policyTokens = _getPolicyTokensOperations(this._client);
    this.policySetDefinitionVersions = _getPolicySetDefinitionVersionsOperations(this._client);
    this.policySetDefinitions = _getPolicySetDefinitionsOperations(this._client);
    this.policyDefinitionVersions = _getPolicyDefinitionVersionsOperations(this._client);
    this.policyDefinitions = _getPolicyDefinitionsOperations(this._client);
    this.policyAssignments = _getPolicyAssignmentsOperations(this._client);
  }
  /** The operation groups for policyTokens */
  policyTokens;
  /** The operation groups for policySetDefinitionVersions */
  policySetDefinitionVersions;
  /** The operation groups for policySetDefinitions */
  policySetDefinitions;
  /** The operation groups for policyDefinitionVersions */
  policyDefinitionVersions;
  /** The operation groups for policyDefinitions */
  policyDefinitions;
  /** The operation groups for policyAssignments */
  policyAssignments;
}
export {
  PolicyClient,
  PolicyClientOptionalParams
};
