// @azure/notification-hubs/dist/commonjs/api/beginSubmitNotificationHubJob.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var beginSubmitNotificationHubJob_exports = {};
__export(beginSubmitNotificationHubJob_exports, {
  beginSubmitNotificationHubJob: () => beginSubmitNotificationHubJob
});
module.exports = __toCommonJS(beginSubmitNotificationHubJob_exports);
var import_core_util = require("@azure/core-util");
var import_getNotificationHubJob = require("./getNotificationHubJob.js");
var import_submitNotificationHubJob = require("./submitNotificationHubJob.js");
async function beginSubmitNotificationHubJob(context, notificationHubJob, polledOperationOptions = {}) {
  let submittedJob = await (0, import_submitNotificationHubJob.submitNotificationHubJob)(
    context,
    notificationHubJob,
    polledOperationOptions
  );
  const state = {
    status: "notStarted"
  };
  const progressCallbacks = /* @__PURE__ */ new Map();
  const processProgressCallbacks = async () => progressCallbacks.forEach((h) => h(state));
  let resultPromise;
  const abortController = new AbortController();
  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2e3;
  const poller = {
    async poll(options) {
      submittedJob = await (0, import_getNotificationHubJob.getNotificationHubJob)(context, submittedJob.jobId, options);
      if (submittedJob.status === "Running" || submittedJob.status === "Started") {
        state.status = "running";
      }
      if (submittedJob.status === "Completed") {
        state.status = "succeeded";
        state.result = submittedJob;
      }
      if (submittedJob.status === "Failed") {
        state.status = "failed";
        state.error = new Error(submittedJob.failure);
      }
      await processProgressCallbacks();
      if (state.status === "canceled") {
        throw new Error("Operation was canceled");
      }
      if (state.status === "failed") {
        throw state.error;
      }
      return state;
    },
    pollUntilDone(pollOptions) {
      return resultPromise ??= (async () => {
        const { abortSignal: inputAbortSignal } = pollOptions || {};
        function abortListener() {
          abortController.abort();
        }
        const abortSignal = abortController.signal;
        if (inputAbortSignal?.aborted) {
          abortController.abort();
        } else if (!abortSignal.aborted) {
          inputAbortSignal?.addEventListener("abort", abortListener, { once: true });
        }
        try {
          if (!poller.isDone) {
            await poller.poll({ abortSignal });
            while (!poller.isDone) {
              await (0, import_core_util.delay)(currentPollIntervalInMs, { abortSignal });
              await poller.poll({ abortSignal });
            }
          }
        } finally {
          inputAbortSignal?.removeEventListener("abort", abortListener);
        }
        switch (state.status) {
          case "succeeded":
            return poller.result;
          case "canceled":
            throw new Error("Operation was canceled");
          case "failed":
            throw state.error;
          case "notStarted":
          case "running":
            throw new Error(`Polling completed without succeeding or failing`);
        }
      })().finally(() => {
        resultPromise = void 0;
      });
    },
    onProgress(callback) {
      const s = /* @__PURE__ */ Symbol();
      progressCallbacks.set(s, callback);
      return () => progressCallbacks.delete(s);
    },
    get isDone() {
      return ["succeeded", "failed", "canceled"].includes(state.status);
    },
    get operationState() {
      return state;
    },
    get result() {
      return state.result;
    },
    async serialize() {
      return JSON.stringify({ state });
    },
    async submitted() {
      return;
    },
    then(onfulfilled, onrejected) {
      return poller.pollUntilDone().then(onfulfilled, onrejected);
    },
    catch(onrejected) {
      return poller.pollUntilDone().catch(onrejected);
    },
    finally(onfinally) {
      return poller.pollUntilDone().finally(onfinally);
    },
    [Symbol.toStringTag]: "Poller"
  };
  return poller;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  beginSubmitNotificationHubJob
});
