// @azure/notification-hubs/dist/commonjs/api/clientContext.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var clientContext_exports = {};
__export(clientContext_exports, {
  createClientContext: () => createClientContext
});
module.exports = __toCommonJS(clientContext_exports);
var constants = __toESM(require("../utils/constants.js"));
var import_core_rest_pipeline = require("@azure/core-rest-pipeline");
var import_connectionStringUtils = require("../auth/connectionStringUtils.js");
var import_core_client = require("@azure-rest/core-client");
const API_VERSION = "2020-06";
function createClientContext(connectionString, hubName, options = {}) {
  return new NotificationHubsServiceClient(connectionString, hubName, options);
}
class NotificationHubsServiceClient {
  sasTokenCredential;
  baseUrl;
  hubName;
  client;
  httpClient;
  constructor(connectionString, hubName, options = {}) {
    this.hubName = hubName;
    const parsedConnection = (0, import_connectionStringUtils.parseNotificationHubsConnectionString)(connectionString);
    this.baseUrl = parsedConnection.endpoint.replace("sb://", "https://");
    this.sasTokenCredential = (0, import_connectionStringUtils.createTokenCredentialFromConnection)(
      parsedConnection.sharedAccessKey,
      parsedConnection.sharedAccessKeyName
    );
    const packageDetails = `azsdk-js-notificationhubs/${constants.SDK_VERSION}`;
    const userAgentPrefix = options.userAgentOptions?.userAgentPrefix ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}` : `${packageDetails}`;
    this.httpClient = options?.httpClient ?? (0, import_core_rest_pipeline.createDefaultHttpClient)();
    this.client = (0, import_core_client.getClient)(this.baseUrl, {
      userAgentOptions: {
        userAgentPrefix
      },
      ...options
    });
  }
  async createHeaders(operationName, rawHeaders) {
    const authorization = await this.sasTokenCredential.getToken(this.baseUrl);
    if (!authorization) {
      throw new import_core_rest_pipeline.RestError("Failed to get the authorization header", { statusCode: 401 });
    }
    const headers = (0, import_core_rest_pipeline.createHttpHeaders)(rawHeaders);
    headers.set("Authorization", authorization.token);
    headers.set("x-ms-version", API_VERSION);
    headers.set(
      "x-ms-azsdk-telemetry",
      `class=NotificationHubsServiceClient;method=${operationName}`
    );
    return headers;
  }
  sendRequest(request) {
    return this.client.pipeline.sendRequest(this.httpClient, request);
  }
  requestUrl() {
    const url = new URL(this.baseUrl);
    url.pathname = this.hubName;
    url.searchParams.set("api-version", API_VERSION);
    return url;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createClientContext
});
