// @azure/arm-policy/dist/commonjs/api/policyDefinitions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyDefinitions_exports = {};
__export(policyDefinitions_exports, {
  $delete: () => import_operations.$delete,
  PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionsCreateOrUpdateOptionalParams: () => import_options.PolicyDefinitionsCreateOrUpdateOptionalParams,
  PolicyDefinitionsDeleteAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionsDeleteOptionalParams: () => import_options.PolicyDefinitionsDeleteOptionalParams,
  PolicyDefinitionsGetAtManagementGroupOptionalParams: () => import_options.PolicyDefinitionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionsGetBuiltInOptionalParams: () => import_options.PolicyDefinitionsGetBuiltInOptionalParams,
  PolicyDefinitionsGetOptionalParams: () => import_options.PolicyDefinitionsGetOptionalParams,
  PolicyDefinitionsListBuiltInOptionalParams: () => import_options.PolicyDefinitionsListBuiltInOptionalParams,
  PolicyDefinitionsListByManagementGroupOptionalParams: () => import_options.PolicyDefinitionsListByManagementGroupOptionalParams,
  PolicyDefinitionsListOptionalParams: () => import_options.PolicyDefinitionsListOptionalParams,
  createOrUpdate: () => import_operations.createOrUpdate,
  createOrUpdateAtManagementGroup: () => import_operations.createOrUpdateAtManagementGroup,
  deleteAtManagementGroup: () => import_operations.deleteAtManagementGroup,
  get: () => import_operations.get,
  getAtManagementGroup: () => import_operations.getAtManagementGroup,
  getBuiltIn: () => import_operations.getBuiltIn,
  list: () => import_operations.list,
  listBuiltIn: () => import_operations.listBuiltIn,
  listByManagementGroup: () => import_operations.listByManagementGroup
});
module.exports = __toCommonJS(policyDefinitions_exports);
var import_operations = require("./operations.js");
var import_options = require("./options.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionsCreateOrUpdateOptionalParams,
  PolicyDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionsDeleteOptionalParams,
  PolicyDefinitionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionsGetBuiltInOptionalParams,
  PolicyDefinitionsGetOptionalParams,
  PolicyDefinitionsListBuiltInOptionalParams,
  PolicyDefinitionsListByManagementGroupOptionalParams,
  PolicyDefinitionsListOptionalParams,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listBuiltIn,
  listByManagementGroup
});
