// @azure/arm-policy/dist/commonjs/api/policySetDefinitions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policySetDefinitions_exports = {};
__export(policySetDefinitions_exports, {
  $delete: () => import_operations.$delete,
  PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionsCreateOrUpdateOptionalParams: () => import_options.PolicySetDefinitionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionsDeleteAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionsDeleteOptionalParams: () => import_options.PolicySetDefinitionsDeleteOptionalParams,
  PolicySetDefinitionsGetAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionsGetBuiltInOptionalParams: () => import_options.PolicySetDefinitionsGetBuiltInOptionalParams,
  PolicySetDefinitionsGetOptionalParams: () => import_options.PolicySetDefinitionsGetOptionalParams,
  PolicySetDefinitionsListBuiltInOptionalParams: () => import_options.PolicySetDefinitionsListBuiltInOptionalParams,
  PolicySetDefinitionsListByManagementGroupOptionalParams: () => import_options.PolicySetDefinitionsListByManagementGroupOptionalParams,
  PolicySetDefinitionsListOptionalParams: () => import_options.PolicySetDefinitionsListOptionalParams,
  createOrUpdate: () => import_operations.createOrUpdate,
  createOrUpdateAtManagementGroup: () => import_operations.createOrUpdateAtManagementGroup,
  deleteAtManagementGroup: () => import_operations.deleteAtManagementGroup,
  get: () => import_operations.get,
  getAtManagementGroup: () => import_operations.getAtManagementGroup,
  getBuiltIn: () => import_operations.getBuiltIn,
  list: () => import_operations.list,
  listBuiltIn: () => import_operations.listBuiltIn,
  listByManagementGroup: () => import_operations.listByManagementGroup
});
module.exports = __toCommonJS(policySetDefinitions_exports);
var import_operations = require("./operations.js");
var import_options = require("./options.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionsDeleteOptionalParams,
  PolicySetDefinitionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionsGetBuiltInOptionalParams,
  PolicySetDefinitionsGetOptionalParams,
  PolicySetDefinitionsListBuiltInOptionalParams,
  PolicySetDefinitionsListByManagementGroupOptionalParams,
  PolicySetDefinitionsListOptionalParams,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listBuiltIn,
  listByManagementGroup
});
