// @azure/arm-policy/dist/commonjs/classic/policySetDefinitions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policySetDefinitions_exports = {};
__export(policySetDefinitions_exports, {
  _getPolicySetDefinitionsOperations: () => _getPolicySetDefinitionsOperations
});
module.exports = __toCommonJS(policySetDefinitions_exports);
var import_operations = require("../../api/policySetDefinitions/operations.js");
function _getPolicySetDefinitions(context) {
  return {
    listByManagementGroup: (managementGroupId, options) => (0, import_operations.listByManagementGroup)(context, managementGroupId, options),
    deleteAtManagementGroup: (managementGroupId, policySetDefinitionName, options) => (0, import_operations.deleteAtManagementGroup)(context, managementGroupId, policySetDefinitionName, options),
    createOrUpdateAtManagementGroup: (managementGroupId, policySetDefinitionName, parameters, options) => (0, import_operations.createOrUpdateAtManagementGroup)(
      context,
      managementGroupId,
      policySetDefinitionName,
      parameters,
      options
    ),
    getAtManagementGroup: (managementGroupId, policySetDefinitionName, options) => (0, import_operations.getAtManagementGroup)(context, managementGroupId, policySetDefinitionName, options),
    listBuiltIn: (options) => (0, import_operations.listBuiltIn)(context, options),
    getBuiltIn: (policySetDefinitionName, options) => (0, import_operations.getBuiltIn)(context, policySetDefinitionName, options),
    list: (options) => (0, import_operations.list)(context, options),
    delete: (policySetDefinitionName, options) => (0, import_operations.$delete)(context, policySetDefinitionName, options),
    createOrUpdate: (policySetDefinitionName, parameters, options) => (0, import_operations.createOrUpdate)(context, policySetDefinitionName, parameters, options),
    get: (policySetDefinitionName, options) => (0, import_operations.get)(context, policySetDefinitionName, options)
  };
}
function _getPolicySetDefinitionsOperations(context) {
  return {
    ..._getPolicySetDefinitions(context)
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  _getPolicySetDefinitionsOperations
});
