// @azure/arm-policy/dist/commonjs/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var src_exports = {};
__export(src_exports, {
  AssignmentType: () => import_models.AssignmentType,
  AzureClouds: () => import_cloudSettingHelpers.AzureClouds,
  AzureSupportedClouds: () => import_cloudSettingHelpers.AzureSupportedClouds,
  ContinuablePage: () => import_pagingHelpers.ContinuablePage,
  CreatedByType: () => import_models.CreatedByType,
  EnforcementMode: () => import_models.EnforcementMode,
  ErrorAdditionalInfo: () => import_models.ErrorAdditionalInfo,
  ErrorDetail: () => import_models.ErrorDetail,
  ErrorResponse: () => import_models.ErrorResponse,
  ExtensionResource: () => import_models.ExtensionResource,
  ExternalEndpointResult: () => import_models.ExternalEndpointResult,
  ExternalEvaluationEndpointInvocationResult: () => import_models.ExternalEvaluationEndpointInvocationResult,
  ExternalEvaluationEndpointSettings: () => import_models.ExternalEvaluationEndpointSettings,
  ExternalEvaluationEnforcementSettings: () => import_models.ExternalEvaluationEnforcementSettings,
  Identity: () => import_models.Identity,
  KnownAssignmentType: () => import_models.KnownAssignmentType,
  KnownCreatedByType: () => import_models.KnownCreatedByType,
  KnownEnforcementMode: () => import_models.KnownEnforcementMode,
  KnownExternalEndpointResult: () => import_models.KnownExternalEndpointResult,
  KnownOverrideKind: () => import_models.KnownOverrideKind,
  KnownParameterType: () => import_models.KnownParameterType,
  KnownPolicyTokenResult: () => import_models.KnownPolicyTokenResult,
  KnownPolicyType: () => import_models.KnownPolicyType,
  KnownSelectorKind: () => import_models.KnownSelectorKind,
  KnownVersions: () => import_models.KnownVersions,
  NonComplianceMessage: () => import_models.NonComplianceMessage,
  Override: () => import_models.Override,
  OverrideKind: () => import_models.OverrideKind,
  PageSettings: () => import_pagingHelpers.PageSettings,
  PagedAsyncIterableIterator: () => import_pagingHelpers.PagedAsyncIterableIterator,
  ParameterDefinitionsValue: () => import_models.ParameterDefinitionsValue,
  ParameterDefinitionsValueMetadata: () => import_models.ParameterDefinitionsValueMetadata,
  ParameterType: () => import_models.ParameterType,
  ParameterValuesValue: () => import_models.ParameterValuesValue,
  PolicyAssignment: () => import_models.PolicyAssignment,
  PolicyAssignmentProperties: () => import_models.PolicyAssignmentProperties,
  PolicyAssignmentUpdate: () => import_models.PolicyAssignmentUpdate,
  PolicyAssignmentUpdateProperties: () => import_models.PolicyAssignmentUpdateProperties,
  PolicyAssignmentsCreateOptionalParams: () => import_policyAssignments.PolicyAssignmentsCreateOptionalParams,
  PolicyAssignmentsDeleteOptionalParams: () => import_policyAssignments.PolicyAssignmentsDeleteOptionalParams,
  PolicyAssignmentsGetOptionalParams: () => import_policyAssignments.PolicyAssignmentsGetOptionalParams,
  PolicyAssignmentsListForManagementGroupOptionalParams: () => import_policyAssignments.PolicyAssignmentsListForManagementGroupOptionalParams,
  PolicyAssignmentsListForResourceGroupOptionalParams: () => import_policyAssignments.PolicyAssignmentsListForResourceGroupOptionalParams,
  PolicyAssignmentsListForResourceOptionalParams: () => import_policyAssignments.PolicyAssignmentsListForResourceOptionalParams,
  PolicyAssignmentsListOptionalParams: () => import_policyAssignments.PolicyAssignmentsListOptionalParams,
  PolicyAssignmentsOperations: () => import_classic.PolicyAssignmentsOperations,
  PolicyAssignmentsUpdateOptionalParams: () => import_policyAssignments.PolicyAssignmentsUpdateOptionalParams,
  PolicyClient: () => import_policyClient.PolicyClient,
  PolicyClientOptionalParams: () => import_api.PolicyClientOptionalParams,
  PolicyDefinition: () => import_models.PolicyDefinition,
  PolicyDefinitionGroup: () => import_models.PolicyDefinitionGroup,
  PolicyDefinitionProperties: () => import_models.PolicyDefinitionProperties,
  PolicyDefinitionReference: () => import_models.PolicyDefinitionReference,
  PolicyDefinitionVersion: () => import_models.PolicyDefinitionVersion,
  PolicyDefinitionVersionProperties: () => import_models.PolicyDefinitionVersionProperties,
  PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsCreateOrUpdateOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsDeleteOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsDeleteOptionalParams,
  PolicyDefinitionVersionsGetAtManagementGroupOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsGetBuiltInOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsGetBuiltInOptionalParams,
  PolicyDefinitionVersionsGetOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsGetOptionalParams,
  PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsListAllBuiltinsOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicyDefinitionVersionsListAllOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListAllOptionalParams,
  PolicyDefinitionVersionsListBuiltInOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListBuiltInOptionalParams,
  PolicyDefinitionVersionsListByManagementGroupOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListByManagementGroupOptionalParams,
  PolicyDefinitionVersionsListOptionalParams: () => import_policyDefinitionVersions.PolicyDefinitionVersionsListOptionalParams,
  PolicyDefinitionVersionsOperations: () => import_classic.PolicyDefinitionVersionsOperations,
  PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_policyDefinitions.PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionsCreateOrUpdateOptionalParams: () => import_policyDefinitions.PolicyDefinitionsCreateOrUpdateOptionalParams,
  PolicyDefinitionsDeleteAtManagementGroupOptionalParams: () => import_policyDefinitions.PolicyDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionsDeleteOptionalParams: () => import_policyDefinitions.PolicyDefinitionsDeleteOptionalParams,
  PolicyDefinitionsGetAtManagementGroupOptionalParams: () => import_policyDefinitions.PolicyDefinitionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionsGetBuiltInOptionalParams: () => import_policyDefinitions.PolicyDefinitionsGetBuiltInOptionalParams,
  PolicyDefinitionsGetOptionalParams: () => import_policyDefinitions.PolicyDefinitionsGetOptionalParams,
  PolicyDefinitionsListBuiltInOptionalParams: () => import_policyDefinitions.PolicyDefinitionsListBuiltInOptionalParams,
  PolicyDefinitionsListByManagementGroupOptionalParams: () => import_policyDefinitions.PolicyDefinitionsListByManagementGroupOptionalParams,
  PolicyDefinitionsListOptionalParams: () => import_policyDefinitions.PolicyDefinitionsListOptionalParams,
  PolicyDefinitionsOperations: () => import_classic.PolicyDefinitionsOperations,
  PolicyLogInfo: () => import_models.PolicyLogInfo,
  PolicySetDefinition: () => import_models.PolicySetDefinition,
  PolicySetDefinitionProperties: () => import_models.PolicySetDefinitionProperties,
  PolicySetDefinitionVersion: () => import_models.PolicySetDefinitionVersion,
  PolicySetDefinitionVersionProperties: () => import_models.PolicySetDefinitionVersionProperties,
  PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsCreateOrUpdateOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsDeleteOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsDeleteOptionalParams,
  PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsGetBuiltInOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsGetBuiltInOptionalParams,
  PolicySetDefinitionVersionsGetOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsGetOptionalParams,
  PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListAllBuiltinsOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicySetDefinitionVersionsListAllOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListAllOptionalParams,
  PolicySetDefinitionVersionsListBuiltInOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListBuiltInOptionalParams,
  PolicySetDefinitionVersionsListByManagementGroupOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListByManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListOptionalParams: () => import_policySetDefinitionVersions.PolicySetDefinitionVersionsListOptionalParams,
  PolicySetDefinitionVersionsOperations: () => import_classic.PolicySetDefinitionVersionsOperations,
  PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionsCreateOrUpdateOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionsDeleteAtManagementGroupOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionsDeleteOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsDeleteOptionalParams,
  PolicySetDefinitionsGetAtManagementGroupOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionsGetBuiltInOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsGetBuiltInOptionalParams,
  PolicySetDefinitionsGetOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsGetOptionalParams,
  PolicySetDefinitionsListBuiltInOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsListBuiltInOptionalParams,
  PolicySetDefinitionsListByManagementGroupOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsListByManagementGroupOptionalParams,
  PolicySetDefinitionsListOptionalParams: () => import_policySetDefinitions.PolicySetDefinitionsListOptionalParams,
  PolicySetDefinitionsOperations: () => import_classic.PolicySetDefinitionsOperations,
  PolicyTokenOperation: () => import_models.PolicyTokenOperation,
  PolicyTokenRequest: () => import_models.PolicyTokenRequest,
  PolicyTokenResponse: () => import_models.PolicyTokenResponse,
  PolicyTokenResult: () => import_models.PolicyTokenResult,
  PolicyTokensAcquireAtManagementGroupOptionalParams: () => import_policyTokens.PolicyTokensAcquireAtManagementGroupOptionalParams,
  PolicyTokensAcquireOptionalParams: () => import_policyTokens.PolicyTokensAcquireOptionalParams,
  PolicyTokensOperations: () => import_classic.PolicyTokensOperations,
  PolicyType: () => import_models.PolicyType,
  ProxyResource: () => import_models.ProxyResource,
  Resource: () => import_models.Resource,
  ResourceIdentityType: () => import_models.ResourceIdentityType,
  ResourceSelector: () => import_models.ResourceSelector,
  Selector: () => import_models.Selector,
  SelectorKind: () => import_models.SelectorKind,
  SystemData: () => import_models.SystemData,
  UserAssignedIdentitiesValue: () => import_models.UserAssignedIdentitiesValue
});
module.exports = __toCommonJS(src_exports);
var import_cloudSettingHelpers = require("./static-helpers/cloudSettingHelpers.js");
var import_pagingHelpers = require("./static-helpers/pagingHelpers.js");
var import_policyClient = require("./policyClient.js");
var import_models = require("./models/index.js");
var import_api = require("./api/index.js");
var import_policyAssignments = require("./api/policyAssignments/index.js");
var import_policyDefinitions = require("./api/policyDefinitions/index.js");
var import_policyDefinitionVersions = require("./api/policyDefinitionVersions/index.js");
var import_policySetDefinitions = require("./api/policySetDefinitions/index.js");
var import_policySetDefinitionVersions = require("./api/policySetDefinitionVersions/index.js");
var import_policyTokens = require("./api/policyTokens/index.js");
var import_classic = require("./classic/index.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AssignmentType,
  AzureClouds,
  AzureSupportedClouds,
  ContinuablePage,
  CreatedByType,
  EnforcementMode,
  ErrorAdditionalInfo,
  ErrorDetail,
  ErrorResponse,
  ExtensionResource,
  ExternalEndpointResult,
  ExternalEvaluationEndpointInvocationResult,
  ExternalEvaluationEndpointSettings,
  ExternalEvaluationEnforcementSettings,
  Identity,
  KnownAssignmentType,
  KnownCreatedByType,
  KnownEnforcementMode,
  KnownExternalEndpointResult,
  KnownOverrideKind,
  KnownParameterType,
  KnownPolicyTokenResult,
  KnownPolicyType,
  KnownSelectorKind,
  KnownVersions,
  NonComplianceMessage,
  Override,
  OverrideKind,
  PageSettings,
  PagedAsyncIterableIterator,
  ParameterDefinitionsValue,
  ParameterDefinitionsValueMetadata,
  ParameterType,
  ParameterValuesValue,
  PolicyAssignment,
  PolicyAssignmentProperties,
  PolicyAssignmentUpdate,
  PolicyAssignmentUpdateProperties,
  PolicyAssignmentsCreateOptionalParams,
  PolicyAssignmentsDeleteOptionalParams,
  PolicyAssignmentsGetOptionalParams,
  PolicyAssignmentsListForManagementGroupOptionalParams,
  PolicyAssignmentsListForResourceGroupOptionalParams,
  PolicyAssignmentsListForResourceOptionalParams,
  PolicyAssignmentsListOptionalParams,
  PolicyAssignmentsOperations,
  PolicyAssignmentsUpdateOptionalParams,
  PolicyClient,
  PolicyClientOptionalParams,
  PolicyDefinition,
  PolicyDefinitionGroup,
  PolicyDefinitionProperties,
  PolicyDefinitionReference,
  PolicyDefinitionVersion,
  PolicyDefinitionVersionProperties,
  PolicyDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicyDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsDeleteOptionalParams,
  PolicyDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsGetBuiltInOptionalParams,
  PolicyDefinitionVersionsGetOptionalParams,
  PolicyDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicyDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicyDefinitionVersionsListAllOptionalParams,
  PolicyDefinitionVersionsListBuiltInOptionalParams,
  PolicyDefinitionVersionsListByManagementGroupOptionalParams,
  PolicyDefinitionVersionsListOptionalParams,
  PolicyDefinitionVersionsOperations,
  PolicyDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicyDefinitionsCreateOrUpdateOptionalParams,
  PolicyDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicyDefinitionsDeleteOptionalParams,
  PolicyDefinitionsGetAtManagementGroupOptionalParams,
  PolicyDefinitionsGetBuiltInOptionalParams,
  PolicyDefinitionsGetOptionalParams,
  PolicyDefinitionsListBuiltInOptionalParams,
  PolicyDefinitionsListByManagementGroupOptionalParams,
  PolicyDefinitionsListOptionalParams,
  PolicyDefinitionsOperations,
  PolicyLogInfo,
  PolicySetDefinition,
  PolicySetDefinitionProperties,
  PolicySetDefinitionVersion,
  PolicySetDefinitionVersionProperties,
  PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsDeleteOptionalParams,
  PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsGetBuiltInOptionalParams,
  PolicySetDefinitionVersionsGetOptionalParams,
  PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicySetDefinitionVersionsListAllOptionalParams,
  PolicySetDefinitionVersionsListBuiltInOptionalParams,
  PolicySetDefinitionVersionsListByManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListOptionalParams,
  PolicySetDefinitionVersionsOperations,
  PolicySetDefinitionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionsDeleteOptionalParams,
  PolicySetDefinitionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionsGetBuiltInOptionalParams,
  PolicySetDefinitionsGetOptionalParams,
  PolicySetDefinitionsListBuiltInOptionalParams,
  PolicySetDefinitionsListByManagementGroupOptionalParams,
  PolicySetDefinitionsListOptionalParams,
  PolicySetDefinitionsOperations,
  PolicyTokenOperation,
  PolicyTokenRequest,
  PolicyTokenResponse,
  PolicyTokenResult,
  PolicyTokensAcquireAtManagementGroupOptionalParams,
  PolicyTokensAcquireOptionalParams,
  PolicyTokensOperations,
  PolicyType,
  ProxyResource,
  Resource,
  ResourceIdentityType,
  ResourceSelector,
  Selector,
  SelectorKind,
  SystemData,
  UserAssignedIdentitiesValue
});
