// @azure/arm-policy/dist/commonjs/policyClient.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyClient_exports = {};
__export(policyClient_exports, {
  PolicyClient: () => PolicyClient,
  PolicyClientOptionalParams: () => import_policyContext.PolicyClientOptionalParams
});
module.exports = __toCommonJS(policyClient_exports);
var import_api = require("./api/index.js");
var import_policyAssignments = require("./classic/policyAssignments/index.js");
var import_policyDefinitionVersions = require("./classic/policyDefinitionVersions/index.js");
var import_policyDefinitions = require("./classic/policyDefinitions/index.js");
var import_policySetDefinitionVersions = require("./classic/policySetDefinitionVersions/index.js");
var import_policySetDefinitions = require("./classic/policySetDefinitions/index.js");
var import_policyTokens = require("./classic/policyTokens/index.js");
var import_policyContext = require("./api/policyContext.js");
class PolicyClient {
  _client;
  /** The pipeline used by this client to make requests */
  pipeline;
  /** To manage and control access to your resources, you can define customized policies and assign them at a scope. */
  constructor(credential, subscriptionIdOrOptions, options) {
    let subscriptionId;
    if (typeof subscriptionIdOrOptions === "string") {
      subscriptionId = subscriptionIdOrOptions;
    } else if (typeof subscriptionIdOrOptions === "object") {
      options = subscriptionIdOrOptions;
    }
    options = options ?? {};
    const prefixFromOptions = options?.userAgentOptions?.userAgentPrefix;
    const userAgentPrefix = prefixFromOptions ? `${prefixFromOptions} azsdk-js-client` : `azsdk-js-client`;
    this._client = (0, import_api.createPolicy)(credential, subscriptionId ?? "", {
      ...options,
      userAgentOptions: { userAgentPrefix }
    });
    this.pipeline = this._client.pipeline;
    this.policyTokens = (0, import_policyTokens._getPolicyTokensOperations)(this._client);
    this.policySetDefinitionVersions = (0, import_policySetDefinitionVersions._getPolicySetDefinitionVersionsOperations)(this._client);
    this.policySetDefinitions = (0, import_policySetDefinitions._getPolicySetDefinitionsOperations)(this._client);
    this.policyDefinitionVersions = (0, import_policyDefinitionVersions._getPolicyDefinitionVersionsOperations)(this._client);
    this.policyDefinitions = (0, import_policyDefinitions._getPolicyDefinitionsOperations)(this._client);
    this.policyAssignments = (0, import_policyAssignments._getPolicyAssignmentsOperations)(this._client);
  }
  /** The operation groups for policyTokens */
  policyTokens;
  /** The operation groups for policySetDefinitionVersions */
  policySetDefinitionVersions;
  /** The operation groups for policySetDefinitions */
  policySetDefinitions;
  /** The operation groups for policyDefinitionVersions */
  policyDefinitionVersions;
  /** The operation groups for policyDefinitions */
  policyDefinitions;
  /** The operation groups for policyAssignments */
  policyAssignments;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PolicyClient,
  PolicyClientOptionalParams
});
