// @azure/arm-policy/dist/commonjs/classic/policyDefinitions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyDefinitions_exports = {};
__export(policyDefinitions_exports, {
  _getPolicyDefinitionsOperations: () => _getPolicyDefinitionsOperations
});
module.exports = __toCommonJS(policyDefinitions_exports);
var import_operations = require("../../api/policyDefinitions/operations.js");
function _getPolicyDefinitions(context) {
  return {
    listByManagementGroup: (managementGroupId, options) => (0, import_operations.listByManagementGroup)(context, managementGroupId, options),
    deleteAtManagementGroup: (managementGroupId, policyDefinitionName, options) => (0, import_operations.deleteAtManagementGroup)(context, managementGroupId, policyDefinitionName, options),
    createOrUpdateAtManagementGroup: (managementGroupId, policyDefinitionName, parameters, options) => (0, import_operations.createOrUpdateAtManagementGroup)(
      context,
      managementGroupId,
      policyDefinitionName,
      parameters,
      options
    ),
    getAtManagementGroup: (managementGroupId, policyDefinitionName, options) => (0, import_operations.getAtManagementGroup)(context, managementGroupId, policyDefinitionName, options),
    listBuiltIn: (options) => (0, import_operations.listBuiltIn)(context, options),
    getBuiltIn: (policyDefinitionName, options) => (0, import_operations.getBuiltIn)(context, policyDefinitionName, options),
    list: (options) => (0, import_operations.list)(context, options),
    delete: (policyDefinitionName, options) => (0, import_operations.$delete)(context, policyDefinitionName, options),
    createOrUpdate: (policyDefinitionName, parameters, options) => (0, import_operations.createOrUpdate)(context, policyDefinitionName, parameters, options),
    get: (policyDefinitionName, options) => (0, import_operations.get)(context, policyDefinitionName, options)
  };
}
function _getPolicyDefinitionsOperations(context) {
  return {
    ..._getPolicyDefinitions(context)
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  _getPolicyDefinitionsOperations
});
