// @azure/notification-hubs/dist/commonjs/utils/constants.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var constants_exports = {};
__export(constants_exports, {
  JSON_CONTENT_TYPE: () => JSON_CONTENT_TYPE,
  SDK_VERSION: () => SDK_VERSION,
  STREAM_CONTENT_TYPE: () => STREAM_CONTENT_TYPE,
  WNS_BADGE: () => WNS_BADGE,
  WNS_RAW: () => WNS_RAW,
  WNS_TITLE: () => WNS_TITLE,
  WNS_TOAST: () => WNS_TOAST,
  WNS_TYPE_NAME: () => WNS_TYPE_NAME,
  XML_CONTENT_TYPE: () => XML_CONTENT_TYPE
});
module.exports = __toCommonJS(constants_exports);
const SDK_VERSION = "2.1.0";
const JSON_CONTENT_TYPE = "application/json;charset=utf-8";
const XML_CONTENT_TYPE = "application/xml";
const STREAM_CONTENT_TYPE = "application/octet-stream";
const WNS_TYPE_NAME = "X-WNS-Type";
const WNS_RAW = "wns/raw";
const WNS_BADGE = "wns/badge";
const WNS_TITLE = "wns/tile";
const WNS_TOAST = "wns/toast";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  JSON_CONTENT_TYPE,
  SDK_VERSION,
  STREAM_CONTENT_TYPE,
  WNS_BADGE,
  WNS_RAW,
  WNS_TITLE,
  WNS_TOAST,
  WNS_TYPE_NAME,
  XML_CONTENT_TYPE
});
