// @azure/arm-policy/dist/commonjs/api/policyAssignments/operations.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var operations_exports = {};
__export(operations_exports, {
  $delete: () => $delete,
  _$deleteDeserialize: () => _$deleteDeserialize,
  _$deleteSend: () => _$deleteSend,
  _createDeserialize: () => _createDeserialize,
  _createSend: () => _createSend,
  _getDeserialize: () => _getDeserialize,
  _getSend: () => _getSend,
  _listDeserialize: () => _listDeserialize,
  _listForManagementGroupDeserialize: () => _listForManagementGroupDeserialize,
  _listForManagementGroupSend: () => _listForManagementGroupSend,
  _listForResourceDeserialize: () => _listForResourceDeserialize,
  _listForResourceGroupDeserialize: () => _listForResourceGroupDeserialize,
  _listForResourceGroupSend: () => _listForResourceGroupSend,
  _listForResourceSend: () => _listForResourceSend,
  _listSend: () => _listSend,
  _updateDeserialize: () => _updateDeserialize,
  _updateSend: () => _updateSend,
  create: () => create,
  get: () => get,
  list: () => list,
  listForManagementGroup: () => listForManagementGroup,
  listForResource: () => listForResource,
  listForResourceGroup: () => listForResourceGroup,
  update: () => update
});
module.exports = __toCommonJS(operations_exports);
var import_models = require("../../models/models.js");
var import_pagingHelpers = require("../../static-helpers/pagingHelpers.js");
var import_urlTemplate = require("../../static-helpers/urlTemplate.js");
var import_core_client = require("@azure-rest/core-client");
function _listForResourceSend(context, resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{+parentResourcePath}/{+resourceType}/{resourceName}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      resourceGroupName,
      resourceProviderNamespace,
      parentResourcePath,
      resourceType,
      resourceName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForResourceDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyAssignmentListResultDeserializer)(result.body);
}
function listForResource(context, resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listForResourceSend(
      context,
      resourceGroupName,
      resourceProviderNamespace,
      parentResourcePath,
      resourceType,
      resourceName,
      options
    ),
    _listForResourceDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listSend(context, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyAssignmentListResultDeserializer)(result.body);
}
function list(context, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listSend(context, options),
    _listDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listForManagementGroupSend(context, managementGroupId, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      managementGroupId,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForManagementGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyAssignmentListResultDeserializer)(result.body);
}
function listForManagementGroup(context, managementGroupId, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listForManagementGroupSend(context, managementGroupId, options),
    _listForManagementGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _listForResourceGroupSend(context, resourceGroupName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Authorization/policyAssignments{?api%2Dversion,%24filter,%24expand,%24top}",
    {
      subscriptionId: context.subscriptionId,
      resourceGroupName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24filter": options?.filter,
      "%24expand": options?.expand,
      "%24top": options?.top
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _listForResourceGroupDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models._policyAssignmentListResultDeserializer)(result.body);
}
function listForResourceGroup(context, resourceGroupName, options = { requestOptions: {} }) {
  return (0, import_pagingHelpers.buildPagedAsyncIterator)(
    context,
    () => _listForResourceGroupSend(context, resourceGroupName, options),
    _listForResourceGroupDeserialize,
    ["200"],
    { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" }
  );
}
function _$deleteSend(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).delete({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _$deleteDeserialize(result) {
  const expectedStatuses = ["200", "204"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyAssignmentDeserializer)(result.body);
}
async function $delete(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const result = await _$deleteSend(context, scope, policyAssignmentName, options);
  return _$deleteDeserialize(result);
}
function _updateSend(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).patch({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: (0, import_models.policyAssignmentUpdateSerializer)(parameters)
  });
}
async function _updateDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyAssignmentDeserializer)(result.body);
}
async function update(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const result = await _updateSend(context, scope, policyAssignmentName, parameters, options);
  return _updateDeserialize(result);
}
function _createSend(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01"
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).put({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    contentType: "application/json",
    headers: { accept: "application/json", ...options.requestOptions?.headers },
    body: (0, import_models.policyAssignmentSerializer)(parameters)
  });
}
async function _createDeserialize(result) {
  const expectedStatuses = ["201"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyAssignmentDeserializer)(result.body);
}
async function create(context, scope, policyAssignmentName, parameters, options = { requestOptions: {} }) {
  const result = await _createSend(context, scope, policyAssignmentName, parameters, options);
  return _createDeserialize(result);
}
function _getSend(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const path = (0, import_urlTemplate.expandUrlTemplate)(
    "/{+scope}/providers/Microsoft.Authorization/policyAssignments/{policyAssignmentName}{?api%2Dversion,%24expand}",
    {
      scope,
      policyAssignmentName,
      "api%2Dversion": context.apiVersion ?? "2025-03-01",
      "%24expand": options?.expand
    },
    {
      allowReserved: options?.requestOptions?.skipUrlEncoding
    }
  );
  return context.path(path).get({
    ...(0, import_core_client.operationOptionsToRequestParameters)(options),
    headers: { accept: "application/json", ...options.requestOptions?.headers }
  });
}
async function _getDeserialize(result) {
  const expectedStatuses = ["200"];
  if (!expectedStatuses.includes(result.status)) {
    const error = (0, import_core_client.createRestError)(result);
    error.details = (0, import_models.errorResponseDeserializer)(result.body);
    throw error;
  }
  return (0, import_models.policyAssignmentDeserializer)(result.body);
}
async function get(context, scope, policyAssignmentName, options = { requestOptions: {} }) {
  const result = await _getSend(context, scope, policyAssignmentName, options);
  return _getDeserialize(result);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  _$deleteDeserialize,
  _$deleteSend,
  _createDeserialize,
  _createSend,
  _getDeserialize,
  _getSend,
  _listDeserialize,
  _listForManagementGroupDeserialize,
  _listForManagementGroupSend,
  _listForResourceDeserialize,
  _listForResourceGroupDeserialize,
  _listForResourceGroupSend,
  _listForResourceSend,
  _listSend,
  _updateDeserialize,
  _updateSend,
  create,
  get,
  list,
  listForManagementGroup,
  listForResource,
  listForResourceGroup,
  update
});
