// @azure/arm-policy/dist/commonjs/api/policySetDefinitionVersions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policySetDefinitionVersions_exports = {};
__export(policySetDefinitionVersions_exports, {
  $delete: () => import_operations.$delete,
  PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsCreateOrUpdateOptionalParams: () => import_options.PolicySetDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsDeleteOptionalParams: () => import_options.PolicySetDefinitionVersionsDeleteOptionalParams,
  PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsGetBuiltInOptionalParams: () => import_options.PolicySetDefinitionVersionsGetBuiltInOptionalParams,
  PolicySetDefinitionVersionsGetOptionalParams: () => import_options.PolicySetDefinitionVersionsGetOptionalParams,
  PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams: () => import_options.PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListAllBuiltinsOptionalParams: () => import_options.PolicySetDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicySetDefinitionVersionsListAllOptionalParams: () => import_options.PolicySetDefinitionVersionsListAllOptionalParams,
  PolicySetDefinitionVersionsListBuiltInOptionalParams: () => import_options.PolicySetDefinitionVersionsListBuiltInOptionalParams,
  PolicySetDefinitionVersionsListByManagementGroupOptionalParams: () => import_options.PolicySetDefinitionVersionsListByManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListOptionalParams: () => import_options.PolicySetDefinitionVersionsListOptionalParams,
  createOrUpdate: () => import_operations.createOrUpdate,
  createOrUpdateAtManagementGroup: () => import_operations.createOrUpdateAtManagementGroup,
  deleteAtManagementGroup: () => import_operations.deleteAtManagementGroup,
  get: () => import_operations.get,
  getAtManagementGroup: () => import_operations.getAtManagementGroup,
  getBuiltIn: () => import_operations.getBuiltIn,
  list: () => import_operations.list,
  listAll: () => import_operations.listAll,
  listAllAtManagementGroup: () => import_operations.listAllAtManagementGroup,
  listAllBuiltins: () => import_operations.listAllBuiltins,
  listBuiltIn: () => import_operations.listBuiltIn,
  listByManagementGroup: () => import_operations.listByManagementGroup
});
module.exports = __toCommonJS(policySetDefinitionVersions_exports);
var import_operations = require("./operations.js");
var import_options = require("./options.js");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  $delete,
  PolicySetDefinitionVersionsCreateOrUpdateAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsCreateOrUpdateOptionalParams,
  PolicySetDefinitionVersionsDeleteAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsDeleteOptionalParams,
  PolicySetDefinitionVersionsGetAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsGetBuiltInOptionalParams,
  PolicySetDefinitionVersionsGetOptionalParams,
  PolicySetDefinitionVersionsListAllAtManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListAllBuiltinsOptionalParams,
  PolicySetDefinitionVersionsListAllOptionalParams,
  PolicySetDefinitionVersionsListBuiltInOptionalParams,
  PolicySetDefinitionVersionsListByManagementGroupOptionalParams,
  PolicySetDefinitionVersionsListOptionalParams,
  createOrUpdate,
  createOrUpdateAtManagementGroup,
  deleteAtManagementGroup,
  get,
  getAtManagementGroup,
  getBuiltIn,
  list,
  listAll,
  listAllAtManagementGroup,
  listAllBuiltins,
  listBuiltIn,
  listByManagementGroup
});
