// @azure/arm-policy/dist/browser/api/policyDefinitionVersions/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { errorResponseDeserializer, policyDefinitionVersionSerializer, policyDefinitionVersionDeserializer, _policyDefinitionVersionListResultDeserializer, } from "../../models/models.js";
import { buildPagedAsyncIterator } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
export function _listAllSend(context, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the policy definition versions for all policy definitions within a subscription. */
export async function listAll(context, options = { requestOptions: {} }) {
    const result = await _listAllSend(context, options);
    return _listAllDeserialize(result);
}
export function _listAllAtManagementGroupSend(context, managementGroupName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the policy definition versions for all policy definitions at the management group scope. */
export async function listAllAtManagementGroup(context, managementGroupName, options = { requestOptions: {} }) {
    const result = await _listAllAtManagementGroupSend(context, managementGroupName, options);
    return _listAllAtManagementGroupDeserialize(result);
}
export function _listAllBuiltinsSend(context, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/listPolicyDefinitionVersions{?api%2Dversion}", {
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllBuiltinsDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the built-in policy definition versions for all built-in policy definitions. */
export async function listAllBuiltins(context, options = { requestOptions: {} }) {
    const result = await _listAllBuiltinsSend(context, options);
    return _listAllBuiltinsDeserialize(result);
}
export function _listByManagementGroupSend(context, managementGroupName, policyDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}", {
        managementGroupName: managementGroupName,
        policyDefinitionName: policyDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listByManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the policy definition versions for the given policy definition in the given management group. */
export function listByManagementGroup(context, managementGroupName, policyDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listByManagementGroupSend(context, managementGroupName, policyDefinitionName, options), _listByManagementGroupDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _deleteAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
export async function _deleteAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200", "204"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return;
}
/** This operation deletes the policy definition in the given management group with the given name. */
export async function deleteAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _deleteAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options);
    return _deleteAtManagementGroupDeserialize(result);
}
export function _createOrUpdateAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: { accept: "application/json", ...options.requestOptions?.headers },
        body: policyDefinitionVersionSerializer(parameters),
    });
}
export async function _createOrUpdateAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200", "201"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policyDefinitionVersionDeserializer(result.body);
}
/** This operation creates or updates a policy definition version in the given management group with the given name. */
export async function createOrUpdateAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options = {
    requestOptions: {},
}) {
    const result = await _createOrUpdateAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options);
    return _createOrUpdateAtManagementGroupDeserialize(result);
}
export function _getAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policyDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the policy definition version in the given management group with the given name. */
export async function getAtManagementGroup(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getAtManagementGroupSend(context, managementGroupName, policyDefinitionName, policyDefinitionVersion, options);
    return _getAtManagementGroupDeserialize(result);
}
export function _listBuiltInSend(context, policyDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}", {
        policyDefinitionName: policyDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listBuiltInDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the built-in policy definition versions for the given policy definition. */
export function listBuiltIn(context, policyDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listBuiltInSend(context, policyDefinitionName, options), _listBuiltInDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _getBuiltInSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getBuiltInDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policyDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the built-in policy definition version with the given name. */
export async function getBuiltIn(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getBuiltInSend(context, policyDefinitionName, policyDefinitionVersion, options);
    return _getBuiltInDeserialize(result);
}
export function _listSend(context, policyDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions{?api%2Dversion,%24top}", {
        subscriptionId: context.subscriptionId,
        policyDefinitionName: policyDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policyDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the policy definition versions for the given policy definition. */
export function list(context, policyDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listSend(context, policyDefinitionName, options), _listDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _$deleteSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
export async function _$deleteDeserialize(result) {
    const expectedStatuses = ["200", "204"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return;
}
/** This operation deletes the policy definition version in the given subscription with the given name. */
/**
 *  @fixme delete is a reserved word that cannot be used as an operation name.
 *         Please add @clientName("clientName") or @clientName("<JS-Specific-Name>", "javascript")
 *         to the operation to override the generated name.
 */
export async function $delete(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _$deleteSend(context, policyDefinitionName, policyDefinitionVersion, options);
    return _$deleteDeserialize(result);
}
export function _createOrUpdateSend(context, policyDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: { accept: "application/json", ...options.requestOptions?.headers },
        body: policyDefinitionVersionSerializer(parameters),
    });
}
export async function _createOrUpdateDeserialize(result) {
    const expectedStatuses = ["200", "201"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policyDefinitionVersionDeserializer(result.body);
}
/** This operation creates or updates a policy definition in the given subscription with the given name. */
export async function createOrUpdate(context, policyDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
    const result = await _createOrUpdateSend(context, policyDefinitionName, policyDefinitionVersion, parameters, options);
    return _createOrUpdateDeserialize(result);
}
export function _getSend(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policyDefinitions/{policyDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        policyDefinitionName: policyDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policyDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the policy definition version in the given subscription with the given name. */
export async function get(context, policyDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getSend(context, policyDefinitionName, policyDefinitionVersion, options);
    return _getDeserialize(result);
}
