// @azure/arm-policy/dist/browser/models/models.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { areAllPropsUndefined } from "../static-helpers/serialization/check-prop-undefined.js";
import { serializeRecord } from "../static-helpers/serialization/serialize-record.js";
export function policyAssignmentSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, [
            "displayName",
            "policyDefinitionId",
            "definitionVersion",
            "notScopes",
            "parameters",
            "description",
            "metadata",
            "enforcementMode",
            "nonComplianceMessages",
            "resourceSelectors",
            "overrides",
            "assignmentType",
        ])
            ? undefined
            : _policyAssignmentPropertiesSerializer(item),
        location: item["location"],
        identity: !item["identity"] ? item["identity"] : identitySerializer(item["identity"]),
    };
}
export function policyAssignmentDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
        ...(!item["properties"]
            ? item["properties"]
            : _policyAssignmentPropertiesDeserializer(item["properties"])),
        location: item["location"],
        identity: !item["identity"] ? item["identity"] : identityDeserializer(item["identity"]),
    };
}
export function policyAssignmentPropertiesSerializer(item) {
    return {
        displayName: item["displayName"],
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        notScopes: !item["notScopes"]
            ? item["notScopes"]
            : item["notScopes"].map((p) => {
                return p;
            }),
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordSerializer(item["parameters"]),
        description: item["description"],
        metadata: item["metadata"],
        enforcementMode: item["enforcementMode"],
        nonComplianceMessages: !item["nonComplianceMessages"]
            ? item["nonComplianceMessages"]
            : nonComplianceMessageArraySerializer(item["nonComplianceMessages"]),
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArraySerializer(item["resourceSelectors"]),
        overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
        assignmentType: item["assignmentType"],
    };
}
export function policyAssignmentPropertiesDeserializer(item) {
    return {
        displayName: item["displayName"],
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        latestDefinitionVersion: item["latestDefinitionVersion"],
        effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
        scope: item["scope"],
        notScopes: !item["notScopes"]
            ? item["notScopes"]
            : item["notScopes"].map((p) => {
                return p;
            }),
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordDeserializer(item["parameters"]),
        description: item["description"],
        metadata: item["metadata"],
        enforcementMode: item["enforcementMode"],
        nonComplianceMessages: !item["nonComplianceMessages"]
            ? item["nonComplianceMessages"]
            : nonComplianceMessageArrayDeserializer(item["nonComplianceMessages"]),
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArrayDeserializer(item["resourceSelectors"]),
        overrides: !item["overrides"]
            ? item["overrides"]
            : overrideArrayDeserializer(item["overrides"]),
        assignmentType: item["assignmentType"],
        instanceId: item["instanceId"],
    };
}
export function parameterValuesValueRecordSerializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : parameterValuesValueSerializer(item[key]);
    });
    return result;
}
export function parameterValuesValueRecordDeserializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : parameterValuesValueDeserializer(item[key]);
    });
    return result;
}
export function parameterValuesValueSerializer(item) {
    return { value: item["value"] };
}
export function parameterValuesValueDeserializer(item) {
    return {
        value: item["value"],
    };
}
/** The policy assignment enforcement mode. Possible values are Default, DoNotEnforce, and Enroll */
export var KnownEnforcementMode;
(function (KnownEnforcementMode) {
    /** The policy effect is enforced during resource creation or update. */
    KnownEnforcementMode["Default"] = "Default";
    /** The policy effect is not enforced during resource creation or update. */
    KnownEnforcementMode["DoNotEnforce"] = "DoNotEnforce";
    /** The policy effect is not enforced during resource creation or update until the resource or scope of the resource is enrolled to the assignment instance. Enrollment occurs upon deployment of the policy enrollment resource. */
    KnownEnforcementMode["Enroll"] = "Enroll";
})(KnownEnforcementMode || (KnownEnforcementMode = {}));
export function nonComplianceMessageArraySerializer(result) {
    return result.map((item) => {
        return nonComplianceMessageSerializer(item);
    });
}
export function nonComplianceMessageArrayDeserializer(result) {
    return result.map((item) => {
        return nonComplianceMessageDeserializer(item);
    });
}
export function nonComplianceMessageSerializer(item) {
    return {
        message: item["message"],
        policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
    };
}
export function nonComplianceMessageDeserializer(item) {
    return {
        message: item["message"],
        policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
    };
}
export function resourceSelectorArraySerializer(result) {
    return result.map((item) => {
        return resourceSelectorSerializer(item);
    });
}
export function resourceSelectorArrayDeserializer(result) {
    return result.map((item) => {
        return resourceSelectorDeserializer(item);
    });
}
export function resourceSelectorSerializer(item) {
    return {
        name: item["name"],
        selectors: !item["selectors"] ? item["selectors"] : selectorArraySerializer(item["selectors"]),
    };
}
export function resourceSelectorDeserializer(item) {
    return {
        name: item["name"],
        selectors: !item["selectors"]
            ? item["selectors"]
            : selectorArrayDeserializer(item["selectors"]),
    };
}
export function selectorArraySerializer(result) {
    return result.map((item) => {
        return selectorSerializer(item);
    });
}
export function selectorArrayDeserializer(result) {
    return result.map((item) => {
        return selectorDeserializer(item);
    });
}
export function selectorSerializer(item) {
    return {
        kind: item["kind"],
        in: !item["in"]
            ? item["in"]
            : item["in"].map((p) => {
                return p;
            }),
        notIn: !item["notIn"]
            ? item["notIn"]
            : item["notIn"].map((p) => {
                return p;
            }),
    };
}
export function selectorDeserializer(item) {
    return {
        kind: item["kind"],
        in: !item["in"]
            ? item["in"]
            : item["in"].map((p) => {
                return p;
            }),
        notIn: !item["notIn"]
            ? item["notIn"]
            : item["notIn"].map((p) => {
                return p;
            }),
    };
}
/** The selector kind. */
export var KnownSelectorKind;
(function (KnownSelectorKind) {
    /** The selector kind to filter policies by the resource location. */
    KnownSelectorKind["ResourceLocation"] = "resourceLocation";
    /** The selector kind to filter policies by the resource type. */
    KnownSelectorKind["ResourceType"] = "resourceType";
    /** The selector kind to filter policies by the resource without location. */
    KnownSelectorKind["ResourceWithoutLocation"] = "resourceWithoutLocation";
    /** The selector kind to filter policies by the policy definition reference ID. */
    KnownSelectorKind["PolicyDefinitionReferenceId"] = "policyDefinitionReferenceId";
})(KnownSelectorKind || (KnownSelectorKind = {}));
export function overrideArraySerializer(result) {
    return result.map((item) => {
        return overrideSerializer(item);
    });
}
export function overrideArrayDeserializer(result) {
    return result.map((item) => {
        return overrideDeserializer(item);
    });
}
export function overrideSerializer(item) {
    return {
        kind: item["kind"],
        value: item["value"],
        selectors: !item["selectors"] ? item["selectors"] : selectorArraySerializer(item["selectors"]),
    };
}
export function overrideDeserializer(item) {
    return {
        kind: item["kind"],
        value: item["value"],
        selectors: !item["selectors"]
            ? item["selectors"]
            : selectorArrayDeserializer(item["selectors"]),
    };
}
/** The override kind. */
export var KnownOverrideKind;
(function (KnownOverrideKind) {
    /** It will override the policy effect type. */
    KnownOverrideKind["PolicyEffect"] = "policyEffect";
    /** It will override the definition version property value of the policy assignment. */
    KnownOverrideKind["DefinitionVersion"] = "definitionVersion";
})(KnownOverrideKind || (KnownOverrideKind = {}));
/** The type of policy assignment. Possible values are NotSpecified, System, SystemHidden, and Custom. Immutable. */
export var KnownAssignmentType;
(function (KnownAssignmentType) {
    /** The not specified assignment type. */
    KnownAssignmentType["NotSpecified"] = "NotSpecified";
    /** The system assignment type. */
    KnownAssignmentType["System"] = "System";
    /** The system hidden assignment type. */
    KnownAssignmentType["SystemHidden"] = "SystemHidden";
    /** The custom assignment type. */
    KnownAssignmentType["Custom"] = "Custom";
})(KnownAssignmentType || (KnownAssignmentType = {}));
export function identitySerializer(item) {
    return {
        type: item["type"],
        userAssignedIdentities: !item["userAssignedIdentities"]
            ? item["userAssignedIdentities"]
            : userAssignedIdentitiesValueRecordSerializer(item["userAssignedIdentities"]),
    };
}
export function identityDeserializer(item) {
    return {
        principalId: item["principalId"],
        tenantId: item["tenantId"],
        type: item["type"],
        userAssignedIdentities: !item["userAssignedIdentities"]
            ? item["userAssignedIdentities"]
            : userAssignedIdentitiesValueRecordDeserializer(item["userAssignedIdentities"]),
    };
}
export function userAssignedIdentitiesValueRecordSerializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : userAssignedIdentitiesValueSerializer(item[key]);
    });
    return result;
}
export function userAssignedIdentitiesValueRecordDeserializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : userAssignedIdentitiesValueDeserializer(item[key]);
    });
    return result;
}
export function userAssignedIdentitiesValueSerializer(item) {
    return item;
}
export function userAssignedIdentitiesValueDeserializer(item) {
    return {
        principalId: item["principalId"],
        clientId: item["clientId"],
    };
}
export function extensionResourceSerializer(item) {
    return item;
}
export function extensionResourceDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
    };
}
export function resourceSerializer(item) {
    return item;
}
export function resourceDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
    };
}
export function systemDataDeserializer(item) {
    return {
        createdBy: item["createdBy"],
        createdByType: item["createdByType"],
        createdAt: !item["createdAt"] ? item["createdAt"] : new Date(item["createdAt"]),
        lastModifiedBy: item["lastModifiedBy"],
        lastModifiedByType: item["lastModifiedByType"],
        lastModifiedAt: !item["lastModifiedAt"]
            ? item["lastModifiedAt"]
            : new Date(item["lastModifiedAt"]),
    };
}
/** The kind of entity that created the resource. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** The entity was created by a user. */
    KnownCreatedByType["User"] = "User";
    /** The entity was created by an application. */
    KnownCreatedByType["Application"] = "Application";
    /** The entity was created by a managed identity. */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** The entity was created by a key. */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
export function errorResponseDeserializer(item) {
    return {
        error: !item["error"] ? item["error"] : errorDetailDeserializer(item["error"]),
    };
}
export function errorDetailDeserializer(item) {
    return {
        code: item["code"],
        message: item["message"],
        target: item["target"],
        details: !item["details"] ? item["details"] : errorDetailArrayDeserializer(item["details"]),
        additionalInfo: !item["additionalInfo"]
            ? item["additionalInfo"]
            : errorAdditionalInfoArrayDeserializer(item["additionalInfo"]),
    };
}
export function errorDetailArrayDeserializer(result) {
    return result.map((item) => {
        return errorDetailDeserializer(item);
    });
}
export function errorAdditionalInfoArrayDeserializer(result) {
    return result.map((item) => {
        return errorAdditionalInfoDeserializer(item);
    });
}
export function errorAdditionalInfoDeserializer(item) {
    return {
        type: item["type"],
        info: item["info"],
    };
}
export function policyAssignmentUpdateSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, ["resourceSelectors", "overrides"])
            ? undefined
            : _policyAssignmentUpdatePropertiesSerializer(item),
        location: item["location"],
        identity: !item["identity"] ? item["identity"] : identitySerializer(item["identity"]),
    };
}
export function policyAssignmentUpdatePropertiesSerializer(item) {
    return {
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArraySerializer(item["resourceSelectors"]),
        overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
    };
}
export function _policyAssignmentListResultDeserializer(item) {
    return {
        value: policyAssignmentArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function policyAssignmentArraySerializer(result) {
    return result.map((item) => {
        return policyAssignmentSerializer(item);
    });
}
export function policyAssignmentArrayDeserializer(result) {
    return result.map((item) => {
        return policyAssignmentDeserializer(item);
    });
}
export function policyDefinitionSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, [
            "policyType",
            "mode",
            "displayName",
            "description",
            "policyRule",
            "metadata",
            "parameters",
            "version",
            "versions",
            "externalEvaluationEnforcementSettings",
        ])
            ? undefined
            : _policyDefinitionPropertiesSerializer(item),
    };
}
export function policyDefinitionDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
        ...(!item["properties"]
            ? item["properties"]
            : _policyDefinitionPropertiesDeserializer(item["properties"])),
    };
}
export function policyDefinitionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsSerializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function policyDefinitionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsDeserializer(item["externalEvaluationEnforcementSettings"]),
    };
}
/** The type of policy definition. Possible values are NotSpecified, BuiltIn, Custom, and Static. */
export var KnownPolicyType;
(function (KnownPolicyType) {
    /** The not specified policy definition type. */
    KnownPolicyType["NotSpecified"] = "NotSpecified";
    /** The built in policy definition type. */
    KnownPolicyType["BuiltIn"] = "BuiltIn";
    /** The custom policy definition type. */
    KnownPolicyType["Custom"] = "Custom";
    /** The static policy definition type. */
    KnownPolicyType["Static"] = "Static";
})(KnownPolicyType || (KnownPolicyType = {}));
export function parameterDefinitionsValueRecordSerializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : parameterDefinitionsValueSerializer(item[key]);
    });
    return result;
}
export function parameterDefinitionsValueRecordDeserializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : parameterDefinitionsValueDeserializer(item[key]);
    });
    return result;
}
export function parameterDefinitionsValueSerializer(item) {
    return {
        type: item["type"],
        allowedValues: !item["allowedValues"]
            ? item["allowedValues"]
            : item["allowedValues"].map((p) => {
                return p;
            }),
        defaultValue: item["defaultValue"],
        schema: item["schema"],
        metadata: !item["metadata"]
            ? item["metadata"]
            : parameterDefinitionsValueMetadataSerializer(item["metadata"]),
    };
}
export function parameterDefinitionsValueDeserializer(item) {
    return {
        type: item["type"],
        allowedValues: !item["allowedValues"]
            ? item["allowedValues"]
            : item["allowedValues"].map((p) => {
                return p;
            }),
        defaultValue: item["defaultValue"],
        schema: item["schema"],
        metadata: !item["metadata"]
            ? item["metadata"]
            : parameterDefinitionsValueMetadataDeserializer(item["metadata"]),
    };
}
/** The data type of the parameter. */
export var KnownParameterType;
(function (KnownParameterType) {
    /** The string parameter type. */
    KnownParameterType["String"] = "String";
    /** The array parameter type. */
    KnownParameterType["Array"] = "Array";
    /** The object parameter type. */
    KnownParameterType["Object"] = "Object";
    /** The boolean parameter type. */
    KnownParameterType["Boolean"] = "Boolean";
    /** The integer parameter type. */
    KnownParameterType["Integer"] = "Integer";
    /** The float parameter type. */
    KnownParameterType["Float"] = "Float";
    /** The date-time parameter type. */
    KnownParameterType["DateTime"] = "DateTime";
})(KnownParameterType || (KnownParameterType = {}));
export function parameterDefinitionsValueMetadataSerializer(item) {
    return {
        ...serializeRecord(item.additionalProperties ?? {}),
        displayName: item["displayName"],
        description: item["description"],
        strongType: item["strongType"],
        assignPermissions: item["assignPermissions"],
    };
}
export function parameterDefinitionsValueMetadataDeserializer(item) {
    return {
        additionalProperties: serializeRecord(item, [
            "displayName",
            "description",
            "strongType",
            "assignPermissions",
        ]),
        displayName: item["displayName"],
        description: item["description"],
        strongType: item["strongType"],
        assignPermissions: item["assignPermissions"],
    };
}
export function externalEvaluationEnforcementSettingsSerializer(item) {
    return {
        missingTokenAction: item["missingTokenAction"],
        resultLifespan: item["resultLifespan"],
        endpointSettings: !item["endpointSettings"]
            ? item["endpointSettings"]
            : externalEvaluationEndpointSettingsSerializer(item["endpointSettings"]),
        roleDefinitionIds: !item["roleDefinitionIds"]
            ? item["roleDefinitionIds"]
            : item["roleDefinitionIds"].map((p) => {
                return p;
            }),
    };
}
export function externalEvaluationEnforcementSettingsDeserializer(item) {
    return {
        missingTokenAction: item["missingTokenAction"],
        resultLifespan: item["resultLifespan"],
        endpointSettings: !item["endpointSettings"]
            ? item["endpointSettings"]
            : externalEvaluationEndpointSettingsDeserializer(item["endpointSettings"]),
        roleDefinitionIds: !item["roleDefinitionIds"]
            ? item["roleDefinitionIds"]
            : item["roleDefinitionIds"].map((p) => {
                return p;
            }),
    };
}
export function externalEvaluationEndpointSettingsSerializer(item) {
    return { kind: item["kind"], details: item["details"] };
}
export function externalEvaluationEndpointSettingsDeserializer(item) {
    return {
        kind: item["kind"],
        details: item["details"],
    };
}
export function proxyResourceSerializer(item) {
    return item;
}
export function proxyResourceDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
    };
}
export function _policyDefinitionListResultDeserializer(item) {
    return {
        value: policyDefinitionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function policyDefinitionArraySerializer(result) {
    return result.map((item) => {
        return policyDefinitionSerializer(item);
    });
}
export function policyDefinitionArrayDeserializer(result) {
    return result.map((item) => {
        return policyDefinitionDeserializer(item);
    });
}
export function policyDefinitionVersionSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, [
            "policyType",
            "mode",
            "displayName",
            "description",
            "policyRule",
            "metadata",
            "parameters",
            "version",
            "externalEvaluationEnforcementSettings",
        ])
            ? undefined
            : _policyDefinitionVersionPropertiesSerializer(item),
    };
}
export function policyDefinitionVersionDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
        ...(!item["properties"]
            ? item["properties"]
            : _policyDefinitionVersionPropertiesDeserializer(item["properties"])),
    };
}
export function policyDefinitionVersionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        version: item["version"],
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsSerializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function policyDefinitionVersionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        version: item["version"],
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsDeserializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function _policyDefinitionVersionListResultDeserializer(item) {
    return {
        value: policyDefinitionVersionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function policyDefinitionVersionArraySerializer(result) {
    return result.map((item) => {
        return policyDefinitionVersionSerializer(item);
    });
}
export function policyDefinitionVersionArrayDeserializer(result) {
    return result.map((item) => {
        return policyDefinitionVersionDeserializer(item);
    });
}
export function policySetDefinitionSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, [
            "policyType",
            "displayName",
            "description",
            "metadata",
            "parameters",
            "policyDefinitions",
            "policyDefinitionGroups",
            "version",
            "versions",
        ])
            ? undefined
            : _policySetDefinitionPropertiesSerializer(item),
    };
}
export function policySetDefinitionDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
        ...(!item["properties"]
            ? item["properties"]
            : _policySetDefinitionPropertiesDeserializer(item["properties"])),
    };
}
export function policySetDefinitionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        policyDefinitions: policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
    };
}
export function policySetDefinitionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        policyDefinitions: policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
    };
}
export function policyDefinitionReferenceArraySerializer(result) {
    return result.map((item) => {
        return policyDefinitionReferenceSerializer(item);
    });
}
export function policyDefinitionReferenceArrayDeserializer(result) {
    return result.map((item) => {
        return policyDefinitionReferenceDeserializer(item);
    });
}
export function policyDefinitionReferenceSerializer(item) {
    return {
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordSerializer(item["parameters"]),
        policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
        groupNames: !item["groupNames"]
            ? item["groupNames"]
            : item["groupNames"].map((p) => {
                return p;
            }),
    };
}
export function policyDefinitionReferenceDeserializer(item) {
    return {
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        latestDefinitionVersion: item["latestDefinitionVersion"],
        effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordDeserializer(item["parameters"]),
        policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
        groupNames: !item["groupNames"]
            ? item["groupNames"]
            : item["groupNames"].map((p) => {
                return p;
            }),
    };
}
export function policyDefinitionGroupArraySerializer(result) {
    return result.map((item) => {
        return policyDefinitionGroupSerializer(item);
    });
}
export function policyDefinitionGroupArrayDeserializer(result) {
    return result.map((item) => {
        return policyDefinitionGroupDeserializer(item);
    });
}
export function policyDefinitionGroupSerializer(item) {
    return {
        name: item["name"],
        displayName: item["displayName"],
        category: item["category"],
        description: item["description"],
        additionalMetadataId: item["additionalMetadataId"],
    };
}
export function policyDefinitionGroupDeserializer(item) {
    return {
        name: item["name"],
        displayName: item["displayName"],
        category: item["category"],
        description: item["description"],
        additionalMetadataId: item["additionalMetadataId"],
    };
}
export function _policySetDefinitionListResultDeserializer(item) {
    return {
        value: policySetDefinitionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function policySetDefinitionArraySerializer(result) {
    return result.map((item) => {
        return policySetDefinitionSerializer(item);
    });
}
export function policySetDefinitionArrayDeserializer(result) {
    return result.map((item) => {
        return policySetDefinitionDeserializer(item);
    });
}
export function policySetDefinitionVersionSerializer(item) {
    return {
        properties: areAllPropsUndefined(item, [
            "policyType",
            "displayName",
            "description",
            "metadata",
            "parameters",
            "policyDefinitions",
            "policyDefinitionGroups",
            "version",
        ])
            ? undefined
            : _policySetDefinitionVersionPropertiesSerializer(item),
    };
}
export function policySetDefinitionVersionDeserializer(item) {
    return {
        id: item["id"],
        name: item["name"],
        type: item["type"],
        systemData: !item["systemData"]
            ? item["systemData"]
            : systemDataDeserializer(item["systemData"]),
        ...(!item["properties"]
            ? item["properties"]
            : _policySetDefinitionVersionPropertiesDeserializer(item["properties"])),
    };
}
export function policySetDefinitionVersionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        policyDefinitions: policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
        version: item["version"],
    };
}
export function policySetDefinitionVersionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        policyDefinitions: policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
        version: item["version"],
    };
}
export function _policySetDefinitionVersionListResultDeserializer(item) {
    return {
        value: policySetDefinitionVersionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function policySetDefinitionVersionArraySerializer(result) {
    return result.map((item) => {
        return policySetDefinitionVersionSerializer(item);
    });
}
export function policySetDefinitionVersionArrayDeserializer(result) {
    return result.map((item) => {
        return policySetDefinitionVersionDeserializer(item);
    });
}
export function policyTokenRequestSerializer(item) {
    return {
        operation: policyTokenOperationSerializer(item["operation"]),
        changeReference: item["changeReference"],
    };
}
export function policyTokenOperationSerializer(item) {
    return { uri: item["uri"], httpMethod: item["httpMethod"], content: item["content"] };
}
export function policyTokenResponseDeserializer(item) {
    return {
        result: item["result"],
        message: item["message"],
        retryAfter: !item["retryAfter"] ? item["retryAfter"] : new Date(item["retryAfter"]),
        results: !item["results"]
            ? item["results"]
            : externalEvaluationEndpointInvocationResultArrayDeserializer(item["results"]),
        changeReference: item["changeReference"],
        token: item["token"],
        tokenId: item["tokenId"],
        expiration: !item["expiration"] ? item["expiration"] : new Date(item["expiration"]),
    };
}
/** The result of the completed token acquisition operation. Possible values are Succeeded and Failed. */
export var KnownPolicyTokenResult;
(function (KnownPolicyTokenResult) {
    /** The token acquisition succeeded. */
    KnownPolicyTokenResult["Succeeded"] = "Succeeded";
    /** The token acquisition failed. */
    KnownPolicyTokenResult["Failed"] = "Failed";
})(KnownPolicyTokenResult || (KnownPolicyTokenResult = {}));
export function externalEvaluationEndpointInvocationResultArrayDeserializer(result) {
    return result.map((item) => {
        return externalEvaluationEndpointInvocationResultDeserializer(item);
    });
}
export function externalEvaluationEndpointInvocationResultDeserializer(item) {
    return {
        policyInfo: !item["policyInfo"]
            ? item["policyInfo"]
            : policyLogInfoDeserializer(item["policyInfo"]),
        result: item["result"],
        message: item["message"],
        retryAfter: !item["retryAfter"] ? item["retryAfter"] : new Date(item["retryAfter"]),
        claims: item["claims"],
        expiration: !item["expiration"] ? item["expiration"] : new Date(item["expiration"]),
    };
}
export function policyLogInfoDeserializer(item) {
    return {
        policyDefinitionId: item["policyDefinitionId"],
        policySetDefinitionId: item["policySetDefinitionId"],
        policyDefinitionReferenceId: item["policyDefinitionReferenceId"],
        policySetDefinitionName: item["policySetDefinitionName"],
        policySetDefinitionDisplayName: item["policySetDefinitionDisplayName"],
        policySetDefinitionVersion: item["policySetDefinitionVersion"],
        policySetDefinitionCategory: item["policySetDefinitionCategory"],
        policyDefinitionName: item["policyDefinitionName"],
        policyDefinitionDisplayName: item["policyDefinitionDisplayName"],
        policyDefinitionVersion: item["policyDefinitionVersion"],
        policyDefinitionEffect: item["policyDefinitionEffect"],
        policyDefinitionGroupNames: !item["policyDefinitionGroupNames"]
            ? item["policyDefinitionGroupNames"]
            : item["policyDefinitionGroupNames"].map((p) => {
                return p;
            }),
        policyAssignmentId: item["policyAssignmentId"],
        policyAssignmentName: item["policyAssignmentName"],
        policyAssignmentDisplayName: item["policyAssignmentDisplayName"],
        policyAssignmentVersion: item["policyAssignmentVersion"],
        policyAssignmentScope: item["policyAssignmentScope"],
        resourceLocation: item["resourceLocation"],
        ancestors: item["ancestors"],
        complianceReasonCode: item["complianceReasonCode"],
        policyExemptionIds: !item["policyExemptionIds"]
            ? item["policyExemptionIds"]
            : item["policyExemptionIds"].map((p) => {
                return p;
            }),
    };
}
/** The result of the external endpoint. Possible values are Succeeded and Failed. */
export var KnownExternalEndpointResult;
(function (KnownExternalEndpointResult) {
    /** The external endpoint succeeded. */
    KnownExternalEndpointResult["Succeeded"] = "Succeeded";
    /** The external endpoint failed. */
    KnownExternalEndpointResult["Failed"] = "Failed";
})(KnownExternalEndpointResult || (KnownExternalEndpointResult = {}));
/** The available API versions. */
export var KnownVersions;
(function (KnownVersions) {
    /** The 2025-03-01 API version. */
    KnownVersions["V20250301"] = "2025-03-01";
})(KnownVersions || (KnownVersions = {}));
export function _policyAssignmentPropertiesSerializer(item) {
    return {
        displayName: item["displayName"],
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        notScopes: !item["notScopes"]
            ? item["notScopes"]
            : item["notScopes"].map((p) => {
                return p;
            }),
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordSerializer(item["parameters"]),
        description: item["description"],
        metadata: item["metadata"],
        enforcementMode: item["enforcementMode"],
        nonComplianceMessages: !item["nonComplianceMessages"]
            ? item["nonComplianceMessages"]
            : nonComplianceMessageArraySerializer(item["nonComplianceMessages"]),
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArraySerializer(item["resourceSelectors"]),
        overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
        assignmentType: item["assignmentType"],
    };
}
export function _policyAssignmentPropertiesDeserializer(item) {
    return {
        displayName: item["displayName"],
        policyDefinitionId: item["policyDefinitionId"],
        definitionVersion: item["definitionVersion"],
        latestDefinitionVersion: item["latestDefinitionVersion"],
        effectiveDefinitionVersion: item["effectiveDefinitionVersion"],
        scope: item["scope"],
        notScopes: !item["notScopes"]
            ? item["notScopes"]
            : item["notScopes"].map((p) => {
                return p;
            }),
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterValuesValueRecordDeserializer(item["parameters"]),
        description: item["description"],
        metadata: item["metadata"],
        enforcementMode: item["enforcementMode"],
        nonComplianceMessages: !item["nonComplianceMessages"]
            ? item["nonComplianceMessages"]
            : nonComplianceMessageArrayDeserializer(item["nonComplianceMessages"]),
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArrayDeserializer(item["resourceSelectors"]),
        overrides: !item["overrides"]
            ? item["overrides"]
            : overrideArrayDeserializer(item["overrides"]),
        assignmentType: item["assignmentType"],
        instanceId: item["instanceId"],
    };
}
export function _policyAssignmentUpdatePropertiesSerializer(item) {
    return {
        resourceSelectors: !item["resourceSelectors"]
            ? item["resourceSelectors"]
            : resourceSelectorArraySerializer(item["resourceSelectors"]),
        overrides: !item["overrides"] ? item["overrides"] : overrideArraySerializer(item["overrides"]),
    };
}
export function _policyDefinitionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsSerializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function _policyDefinitionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsDeserializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function _policyDefinitionVersionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        version: item["version"],
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsSerializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function _policyDefinitionVersionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        mode: item["mode"],
        displayName: item["displayName"],
        description: item["description"],
        policyRule: item["policyRule"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        version: item["version"],
        externalEvaluationEnforcementSettings: !item["externalEvaluationEnforcementSettings"]
            ? item["externalEvaluationEnforcementSettings"]
            : externalEvaluationEnforcementSettingsDeserializer(item["externalEvaluationEnforcementSettings"]),
    };
}
export function _policySetDefinitionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        policyDefinitions: !item["policyDefinitions"]
            ? item["policyDefinitions"]
            : policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
    };
}
export function _policySetDefinitionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        policyDefinitions: !item["policyDefinitions"]
            ? item["policyDefinitions"]
            : policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
        version: item["version"],
        versions: !item["versions"]
            ? item["versions"]
            : item["versions"].map((p) => {
                return p;
            }),
    };
}
export function _policySetDefinitionVersionPropertiesSerializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordSerializer(item["parameters"]),
        policyDefinitions: !item["policyDefinitions"]
            ? item["policyDefinitions"]
            : policyDefinitionReferenceArraySerializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArraySerializer(item["policyDefinitionGroups"]),
        version: item["version"],
    };
}
export function _policySetDefinitionVersionPropertiesDeserializer(item) {
    return {
        policyType: item["policyType"],
        displayName: item["displayName"],
        description: item["description"],
        metadata: item["metadata"],
        parameters: !item["parameters"]
            ? item["parameters"]
            : parameterDefinitionsValueRecordDeserializer(item["parameters"]),
        policyDefinitions: !item["policyDefinitions"]
            ? item["policyDefinitions"]
            : policyDefinitionReferenceArrayDeserializer(item["policyDefinitions"]),
        policyDefinitionGroups: !item["policyDefinitionGroups"]
            ? item["policyDefinitionGroups"]
            : policyDefinitionGroupArrayDeserializer(item["policyDefinitionGroups"]),
        version: item["version"],
    };
}
