// @azure/arm-policy/dist/react-native/api/policySetDefinitionVersions/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { errorResponseDeserializer, policySetDefinitionVersionSerializer, policySetDefinitionVersionDeserializer, _policySetDefinitionVersionListResultDeserializer, } from "../../models/models.js";
import { buildPagedAsyncIterator } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
export function _listAllSend(context, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/listPolicySetDefinitionVersions{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the policy set definition versions for all policy set definitions within a subscription. */
export async function listAll(context, options = { requestOptions: {} }) {
    const result = await _listAllSend(context, options);
    return _listAllDeserialize(result);
}
export function _listAllAtManagementGroupSend(context, managementGroupName, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/listPolicySetDefinitionVersions{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the policy set definition versions for all policy set definitions at the management group scope. */
export async function listAllAtManagementGroup(context, managementGroupName, options = {
    requestOptions: {},
}) {
    const result = await _listAllAtManagementGroupSend(context, managementGroupName, options);
    return _listAllAtManagementGroupDeserialize(result);
}
export function _listAllBuiltinsSend(context, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/listPolicySetDefinitionVersions{?api%2Dversion}", {
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listAllBuiltinsDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation lists all the built-in policy set definition versions for all built-in policy set definitions. */
export async function listAllBuiltins(context, options = { requestOptions: {} }) {
    const result = await _listAllBuiltinsSend(context, options);
    return _listAllBuiltinsDeserialize(result);
}
export function _listByManagementGroupSend(context, managementGroupName, policySetDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions{?api%2Dversion,%24expand,%24top}", {
        managementGroupName: managementGroupName,
        policySetDefinitionName: policySetDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listByManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the policy set definition versions for the given policy set definition in a given management group. */
export function listByManagementGroup(context, managementGroupName, policySetDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listByManagementGroupSend(context, managementGroupName, policySetDefinitionName, options), _listByManagementGroupDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _deleteAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
export async function _deleteAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200", "204"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return;
}
/** This operation deletes the policy set definition version in the given management group with the given name and version. */
export async function deleteAtManagementGroup(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options = {
    requestOptions: {},
}) {
    const result = await _deleteAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options);
    return _deleteAtManagementGroupDeserialize(result);
}
export function _createOrUpdateAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, parameters, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        managementGroupName: managementGroupName,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: { accept: "application/json", ...options.requestOptions?.headers },
        body: policySetDefinitionVersionSerializer(parameters),
    });
}
export async function _createOrUpdateAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200", "201"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policySetDefinitionVersionDeserializer(result.body);
}
/** This operation creates or updates a policy set definition version in the given management group with the given name and version. */
export async function createOrUpdateAtManagementGroup(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, parameters, options = {
    requestOptions: {},
}) {
    const result = await _createOrUpdateAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, parameters, options);
    return _createOrUpdateAtManagementGroupDeserialize(result);
}
export function _getAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Management/managementGroups/{managementGroupName}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion,%24expand}", {
        managementGroupName: managementGroupName,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getAtManagementGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policySetDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the policy set definition version in the given management group with the given name and version. */
export async function getAtManagementGroup(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getAtManagementGroupSend(context, managementGroupName, policySetDefinitionName, policyDefinitionVersion, options);
    return _getAtManagementGroupDeserialize(result);
}
export function _listBuiltInSend(context, policySetDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions{?api%2Dversion,%24expand,%24top}", {
        policySetDefinitionName: policySetDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listBuiltInDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the built-in policy set definition versions for the given built-in policy set definition. */
export function listBuiltIn(context, policySetDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listBuiltInSend(context, policySetDefinitionName, options), _listBuiltInDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _getBuiltInSend(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion,%24expand}", {
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getBuiltInDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policySetDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the built-in policy set definition version with the given name and version. */
export async function getBuiltIn(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getBuiltInSend(context, policySetDefinitionName, policyDefinitionVersion, options);
    return _getBuiltInDeserialize(result);
}
export function _listSend(context, policySetDefinitionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions{?api%2Dversion,%24expand,%24top}", {
        subscriptionId: context.subscriptionId,
        policySetDefinitionName: policySetDefinitionName,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
        "%24top": options?.top,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _listDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _policySetDefinitionVersionListResultDeserializer(result.body);
}
/** This operation retrieves a list of all the policy set definition versions for the given policy set definition. */
export function list(context, policySetDefinitionName, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listSend(context, policySetDefinitionName, options), _listDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink", apiVersion: context.apiVersion ?? "2025-03-01" });
}
export function _$deleteSend(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
export async function _$deleteDeserialize(result) {
    const expectedStatuses = ["200", "204"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return;
}
/** This operation deletes the policy set definition version in the given subscription with the given name and version. */
/**
 *  @fixme delete is a reserved word that cannot be used as an operation name.
 *         Please add @clientName("clientName") or @clientName("<JS-Specific-Name>", "javascript")
 *         to the operation to override the generated name.
 */
export async function $delete(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _$deleteSend(context, policySetDefinitionName, policyDefinitionVersion, options);
    return _$deleteDeserialize(result);
}
export function _createOrUpdateSend(context, policySetDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: { accept: "application/json", ...options.requestOptions?.headers },
        body: policySetDefinitionVersionSerializer(parameters),
    });
}
export async function _createOrUpdateDeserialize(result) {
    const expectedStatuses = ["200", "201"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policySetDefinitionVersionDeserializer(result.body);
}
/** This operation creates or updates a policy set definition version in the given subscription with the given name and version. */
export async function createOrUpdate(context, policySetDefinitionName, policyDefinitionVersion, parameters, options = { requestOptions: {} }) {
    const result = await _createOrUpdateSend(context, policySetDefinitionName, policyDefinitionVersion, parameters, options);
    return _createOrUpdateDeserialize(result);
}
export function _getSend(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{policyDefinitionVersion}{?api%2Dversion,%24expand}", {
        subscriptionId: context.subscriptionId,
        policySetDefinitionName: policySetDefinitionName,
        policyDefinitionVersion: policyDefinitionVersion,
        "api%2Dversion": context.apiVersion ?? "2025-03-01",
        "%24expand": options?.expand,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: { accept: "application/json", ...options.requestOptions?.headers },
    });
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return policySetDefinitionVersionDeserializer(result.body);
}
/** This operation retrieves the policy set definition version in the given subscription with the given name and version. */
export async function get(context, policySetDefinitionName, policyDefinitionVersion, options = { requestOptions: {} }) {
    const result = await _getSend(context, policySetDefinitionName, policyDefinitionVersion, options);
    return _getDeserialize(result);
}
