// @azure/arm-policy/dist/esm/classic/policyDefinitionVersions/index.js
import {
  listAll,
  listAllAtManagementGroup,
  listAllBuiltins,
  listByManagementGroup,
  deleteAtManagementGroup,
  createOrUpdateAtManagementGroup,
  getAtManagementGroup,
  listBuiltIn,
  getBuiltIn,
  list,
  $delete,
  createOrUpdate,
  get
} from "../../api/policyDefinitionVersions/operations.js";
function _getPolicyDefinitionVersions(context) {
  return {
    listAll: (options) => listAll(context, options),
    listAllAtManagementGroup: (managementGroupName, options) => listAllAtManagementGroup(context, managementGroupName, options),
    listAllBuiltins: (options) => listAllBuiltins(context, options),
    listByManagementGroup: (managementGroupName, policyDefinitionName, options) => listByManagementGroup(context, managementGroupName, policyDefinitionName, options),
    deleteAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, options) => deleteAtManagementGroup(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      options
    ),
    createOrUpdateAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options) => createOrUpdateAtManagementGroup(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      parameters,
      options
    ),
    getAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, options) => getAtManagementGroup(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      options
    ),
    listBuiltIn: (policyDefinitionName, options) => listBuiltIn(context, policyDefinitionName, options),
    getBuiltIn: (policyDefinitionName, policyDefinitionVersion, options) => getBuiltIn(context, policyDefinitionName, policyDefinitionVersion, options),
    list: (policyDefinitionName, options) => list(context, policyDefinitionName, options),
    delete: (policyDefinitionName, policyDefinitionVersion, options) => $delete(context, policyDefinitionName, policyDefinitionVersion, options),
    createOrUpdate: (policyDefinitionName, policyDefinitionVersion, parameters, options) => createOrUpdate(context, policyDefinitionName, policyDefinitionVersion, parameters, options),
    get: (policyDefinitionName, policyDefinitionVersion, options) => get(context, policyDefinitionName, policyDefinitionVersion, options)
  };
}
function _getPolicyDefinitionVersionsOperations(context) {
  return {
    ..._getPolicyDefinitionVersions(context)
  };
}
export {
  _getPolicyDefinitionVersionsOperations
};
