// @azure/arm-policy/dist/commonjs/classic/policyDefinitionVersions/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var policyDefinitionVersions_exports = {};
__export(policyDefinitionVersions_exports, {
  _getPolicyDefinitionVersionsOperations: () => _getPolicyDefinitionVersionsOperations
});
module.exports = __toCommonJS(policyDefinitionVersions_exports);
var import_operations = require("../../api/policyDefinitionVersions/operations.js");
function _getPolicyDefinitionVersions(context) {
  return {
    listAll: (options) => (0, import_operations.listAll)(context, options),
    listAllAtManagementGroup: (managementGroupName, options) => (0, import_operations.listAllAtManagementGroup)(context, managementGroupName, options),
    listAllBuiltins: (options) => (0, import_operations.listAllBuiltins)(context, options),
    listByManagementGroup: (managementGroupName, policyDefinitionName, options) => (0, import_operations.listByManagementGroup)(context, managementGroupName, policyDefinitionName, options),
    deleteAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, options) => (0, import_operations.deleteAtManagementGroup)(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      options
    ),
    createOrUpdateAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, parameters, options) => (0, import_operations.createOrUpdateAtManagementGroup)(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      parameters,
      options
    ),
    getAtManagementGroup: (managementGroupName, policyDefinitionName, policyDefinitionVersion, options) => (0, import_operations.getAtManagementGroup)(
      context,
      managementGroupName,
      policyDefinitionName,
      policyDefinitionVersion,
      options
    ),
    listBuiltIn: (policyDefinitionName, options) => (0, import_operations.listBuiltIn)(context, policyDefinitionName, options),
    getBuiltIn: (policyDefinitionName, policyDefinitionVersion, options) => (0, import_operations.getBuiltIn)(context, policyDefinitionName, policyDefinitionVersion, options),
    list: (policyDefinitionName, options) => (0, import_operations.list)(context, policyDefinitionName, options),
    delete: (policyDefinitionName, policyDefinitionVersion, options) => (0, import_operations.$delete)(context, policyDefinitionName, policyDefinitionVersion, options),
    createOrUpdate: (policyDefinitionName, policyDefinitionVersion, parameters, options) => (0, import_operations.createOrUpdate)(context, policyDefinitionName, policyDefinitionVersion, parameters, options),
    get: (policyDefinitionName, policyDefinitionVersion, options) => (0, import_operations.get)(context, policyDefinitionName, policyDefinitionVersion, options)
  };
}
function _getPolicyDefinitionVersionsOperations(context) {
  return {
    ..._getPolicyDefinitionVersions(context)
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  _getPolicyDefinitionVersionsOperations
});
